<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Mary Piper">

<title>Introduction to Differential Gene Expression Analysis - Differential expression of transcripts using Sleuth</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="../../site_libs/quarto-nav/quarto-nav.js"></script>
<script src="../../site_libs/quarto-nav/headroom.min.js"></script>
<script src="../../site_libs/clipboard/clipboard.min.js"></script>
<script src="../../site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="../../site_libs/quarto-search/fuse.min.js"></script>
<script src="../../site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="../../">
<script src="../../site_libs/quarto-html/quarto.js"></script>
<script src="../../site_libs/quarto-html/popper.min.js"></script>
<script src="../../site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="../../site_libs/quarto-html/anchor.min.js"></script>
<link href="../../site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="../../site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="../../site_libs/bootstrap/bootstrap.min.js"></script>
<link href="../../site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="../../site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "navbar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "end",
  "type": "overlay",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


<link rel="stylesheet" href="../../css/styles.css">
</head>

<body class="nav-fixed">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
    <nav class="navbar navbar-expand-lg " data-bs-theme="dark">
      <div class="navbar-container container-fluid">
      <div class="navbar-brand-container mx-auto">
    <a class="navbar-brand" href="../../index.html">
    <span class="navbar-title">Introduction to Differential Gene Expression Analysis</span>
    </a>
  </div>
            <div id="quarto-search" class="" title="Search"></div>
          <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
  <span class="navbar-toggler-icon"></span>
</button>
          <div class="collapse navbar-collapse" id="navbarCollapse">
            <ul class="navbar-nav navbar-nav-scroll me-auto">
  <li class="nav-item">
    <a class="nav-link" href="../../lessons/schedule.html"> 
<span class="menu-text">Schedule</span></a>
  </li>  
</ul>
            <ul class="navbar-nav navbar-nav-scroll ms-auto">
  <li class="nav-item">
    <a class="nav-link" href="https://bioinformatics.sph.harvard.edu/"> 
<span class="menu-text">HBC</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="https://github.com/hbctraining/Intro-to-DGE-Quarto"> 
<span class="menu-text">GitHub</span></a>
  </li>  
  <li class="nav-item">
    <a class="nav-link" href="mailto:hbctraining@hsph.harvard.edu"> 
<span class="menu-text">Contact us</span></a>
  </li>  
</ul>
          </div> <!-- /navcollapse -->
          <div class="quarto-navbar-tools">
</div>
      </div> <!-- /container-fluid -->
    </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article page-navbar">
<!-- sidebar -->
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">On this page</h2>
   
  <ul>
  <li><a href="#learning-objectives" id="toc-learning-objectives" class="nav-link active" data-scroll-target="#learning-objectives">Learning Objectives</a></li>
  <li><a href="#transcript-level-differential-expression" id="toc-transcript-level-differential-expression" class="nav-link" data-scroll-target="#transcript-level-differential-expression">Transcript-level differential expression</a></li>
  <li><a href="#what-is-sleuth" id="toc-what-is-sleuth" class="nav-link" data-scroll-target="#what-is-sleuth">What is Sleuth?</a></li>
  <li><a href="#set-up-for-running-sleuth" id="toc-set-up-for-running-sleuth" class="nav-link" data-scroll-target="#set-up-for-running-sleuth">Set-up for Running Sleuth</a></li>
  <li><a href="#using-wasabi-to-convert-salmon-output-for-sleuth" id="toc-using-wasabi-to-convert-salmon-output-for-sleuth" class="nav-link" data-scroll-target="#using-wasabi-to-convert-salmon-output-for-sleuth">Using Wasabi to convert Salmon output for Sleuth</a></li>
  <li><a href="#sleuth-for-estimation-of-differential-expression-of-transcripts" id="toc-sleuth-for-estimation-of-differential-expression-of-transcripts" class="nav-link" data-scroll-target="#sleuth-for-estimation-of-differential-expression-of-transcripts">Sleuth for estimation of differential expression of transcripts</a></li>
  <li><a href="#sleuth-workflow" id="toc-sleuth-workflow" class="nav-link" data-scroll-target="#sleuth-workflow">Sleuth workflow</a>
  <ul class="collapse">
  <li><a href="#step-1-create-sleuth-object-for-analysis" id="toc-step-1-create-sleuth-object-for-analysis" class="nav-link" data-scroll-target="#step-1-create-sleuth-object-for-analysis">Step 1: Create Sleuth object for analysis</a></li>
  <li><a href="#step-2-fit-the-sleuth-model" id="toc-step-2-fit-the-sleuth-model" class="nav-link" data-scroll-target="#step-2-fit-the-sleuth-model">Step 2: Fit the sleuth model</a></li>
  <li><a href="#step-3-test-significant-differences-between-conditions-using-the-wald-test" id="toc-step-3-test-significant-differences-between-conditions-using-the-wald-test" class="nav-link" data-scroll-target="#step-3-test-significant-differences-between-conditions-using-the-wald-test">Step 3: Test significant differences between conditions using the Wald test</a></li>
  <li><a href="#exploring-transcript-level-expression-between-samples" id="toc-exploring-transcript-level-expression-between-samples" class="nav-link" data-scroll-target="#exploring-transcript-level-expression-between-samples">Exploring transcript-level expression between samples</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<div class="quarto-title-block"><div><h1 class="title">Differential expression of transcripts using Sleuth</h1><button type="button" class="btn code-tools-button" id="quarto-code-tools-source"><i class="bi"></i> Code</button></div></div>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Mary Piper </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">Invalid Date</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<p>Approximate time: 1.5 hours</p>
<section id="learning-objectives" class="level2">
<h2 class="anchored" data-anchor-id="learning-objectives">Learning Objectives</h2>
<ul>
<li>Determine differential expression of isoforms and visualization of results using Sleuth</li>
<li>Understand how Sleuth determines biological and technical variation</li>
</ul>
</section>
<section id="transcript-level-differential-expression" class="level2">
<h2 class="anchored" data-anchor-id="transcript-level-differential-expression">Transcript-level differential expression</h2>
<p>Until this point we have focused on looking for expression changes at the gene-level. However, if you are interested in looking at <strong>splice isoform expression changes between groups</strong> the previous methods (i.e DESeq2) will not work. To demonstrate how to identify transcript-level differential expression we will be using a tool called Sleuth.</p>
<p><img src="../img/workflow-salmon-sleuth.png" width="700"></p>
</section>
<section id="what-is-sleuth" class="level2">
<h2 class="anchored" data-anchor-id="what-is-sleuth">What is Sleuth?</h2>
<p><a href="http://pachterlab.github.io/sleuth/">Sleuth</a> is a fast, lightweight tool that uses transcript abundance estimates output from <strong>pseudo-alignment</strong> algorithms that use <strong>bootstrap sampling</strong>, such as Sailfish, Salmon, and Kallisto, to perform differential expression analysis of gene isoforms.</p>
<p>To analyze the differential expression of gene isoforms, it is expected that RNA-Seq reads will often align to multiple isoforms of the same gene. Therefore, <strong>multimapping reads cannot be ignored</strong> to properly determine abundances of gene isoforms.</p>
<p>Due to the statistical procedure required to assign reads to gene isoforms, in addition to the random processes underlying RNA-Seq, there will be <strong>technical variability in the abundance estimates</strong> output from the pseudo-alignment tool [<a href="https://rawgit.com/pachterlab/sleuth/master/inst/doc/intro.html">2</a>, <a href="https://www.nature.com/articles/nmeth.4324">3</a>]. Therefore, <strong>we would need technical replicates to distinguish technical variability from the biological variability</strong>.</p>
<p>Sleuth accounts for this technical variability by using <strong>bootstraps as a proxy for technical replicates</strong>, which are used to model the technical variability in the abundance estimates. Bootstrapping essentially <strong>calculates the abundance estimates for all genes using a different sub-sample of reads</strong> during each round of bootstrapping. The variation in the abundance estimates output from each round of bootstrapping is used for the estimation of the technical variance for each gene.</p>
<p align="center">
<img src="../img/sleuth_tech_var.png" width="600">
</p>
<p><em>Adapted from: Nature Methods <strong>14</strong>, 687–690 (2017)</em></p>
<p>Sleuth models the unobserved true abundance (logarithm of true counts) using a general linear model, but includes the technical variance (variance between bootstrapping runs) as error in the response variable.</p>
<p align="center">
<img src="../img/sleuth_formula1.png" width="500">
</p>
<p>The observed (log) abundance estimates represent the sum of the true counts and the technical noise. Thereby, sleuth teases apart the source of the variance (technical vs.&nbsp;biological) for estimation of the “true” biological variance when determining whether transcripts are differentially expressed.</p>
<p align="center">
<img src="../img/sleuth_formula2.png" width="425">
</p>
<p>In addition to performing differential expression analysis of transcripts, the sleuth tool also provides an HTML interface allowing exploration of the data and differential expression results interactively. More information about the theory/process for sleuth is available in the <a href="https://www.nature.com/articles/nmeth.4324">Nature Methods paper</a>, this <a href="https://liorpachter.wordpress.com/2015/08/17/a-sleuth-for-rna-seq/">blogpost</a> and step-by-step tutorials are available on the <a href="https://pachterlab.github.io/sleuth/walkthroughs">sleuth website</a>.</p>
<p><strong><em>NOTE:</em></strong> <em>Kallisto is distributed under a non-commercial license, while Sailfish and Salmon are distributed under the <a href="http://www.gnu.org/licenses/gpl.html">GNU General Public License, version 3</a>.</em></p>
</section>
<section id="set-up-for-running-sleuth" class="level2">
<h2 class="anchored" data-anchor-id="set-up-for-running-sleuth">Set-up for Running Sleuth</h2>
<p>Sleuth is a lightweight algorithm that can be quickly run on our personal computers. Let’s <strong>open a new project called <code>DE_sleuth</code></strong>.</p>
<blockquote class="blockquote">
<p><strong>NOTE:</strong> We could have used the previous Salmon project, however we did not include the <code>--numBootstraps 30</code> parameter in our automation script and so we do not have the appropriate files.</p>
</blockquote>
<ol type="1">
<li>Download the Salmon files with bootstraps <a href="https://www.dropbox.com/s/ntpkgkxikb46b35/data.zip?dl=1">from here</a>. Move it into your working directory and decompress the file. You should now see a <code>data</code> directory.</li>
<li>Create a new R script (‘File’ -&gt; ‘New File’ -&gt; ‘Rscript’), and save it as <code>sleuth_de.R</code></li>
</ol>
<p>To perform any analysis, we need to load the libraries for <code>wasabi</code> and <code>sleuth</code>. Sleuth also requires <code>annotables</code>, so this package will be loaded as well:</p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wasabi)</span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sleuth)</span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(annotables)</span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="using-wasabi-to-convert-salmon-output-for-sleuth" class="level2">
<h2 class="anchored" data-anchor-id="using-wasabi-to-convert-salmon-output-for-sleuth">Using Wasabi to convert Salmon output for Sleuth</h2>
<p>Sleuth was built to use the bootstrapped estimates of transcript abundance from Kallisto; however, abundance estimates from Salmon (or Sailfish) work just as well, as long as bootstrapping is performed. To use our Salmon output estimates, we need to convert them to a Sleuth-compatible format using the Wasabi package.</p>
<p>First, we create a simple vector containing the paths to the directories containing the transcript abundance estimates for each sample (folders containing the .quant files). We can use the <code>file.path()</code> function to give the paths to each of the directories.</p>
<p>Now, let’s use this function to create our list of the paths to our transcript abundance files:</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>sf_dirs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="fu">c</span>(<span class="st">"Mov10_kd_2.salmon"</span>, <span class="st">"Mov10_kd_3.salmon"</span>, <span class="st">"Mov10_oe_1.salmon"</span>, <span class="st">"Mov10_oe_2.salmon"</span>, <span class="st">"Mov10_oe_3.salmon"</span>,<span class="st">"Irrel_kd_1.salmon"</span>, <span class="st">"Irrel_kd_2.salmon"</span>, <span class="st">"Irrel_kd_3.salmon"</span>))</span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>sf_dirs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Now, we simply run the <code>prepare_fish_for_sleuth</code> function, which will write some status messages to the console and, when it’s done, each directory will contain an <code>abundance.h5</code> file in a sleuth-compatible format.</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">prepare_fish_for_sleuth</span>(sf_dirs)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Each of the sample directories should now contain the <code>abundance.h5</code> files. These files will be used as input to Sleuth.</p>
<p align="center">
<img src="../img/sleuth_abund_h5.png" width="600">
</p>
</section>
<section id="sleuth-for-estimation-of-differential-expression-of-transcripts" class="level2">
<h2 class="anchored" data-anchor-id="sleuth-for-estimation-of-differential-expression-of-transcripts">Sleuth for estimation of differential expression of transcripts</h2>
<p align="center">
<img src="../img/sleuth_workflow.png" width="700">
</p>
<p>The workflow for Sleuth is similar to the workflow followed for DESeq2, even though, the models for estimating differential expression are very different.</p>
<p><strong>Step 1:</strong> Creation of Sleuth object to provide metadata, estimated counts, and design formula for the analysis, in addition to a <code>annotables</code> database to switch between transcript IDs and associated gene names.</p>
<p><strong>Step 2:</strong> Fit the sleuth model</p>
<ul>
<li><p>Estimation of size (normalization) factors using the median of ratios method (similar to DESeq2)</p></li>
<li><p>Normalization of estimated counts using size factors (est. counts / size factors - similar to DESeq2)</p></li>
<li><p>Filtering of low abundance transcripts (&lt; 5 est counts in more than 47% of the samples)</p></li>
<li><p>Normalization of technical variation estimates</p></li>
<li><p>Estimation of biological variance and shrinkage estimates (With small sample sizes, we will make very bad estimates of transcript-wise dispersion unless we share information across transcripts. Sleuth regularizes the biological variance estimate with shrinkage, similar to DESeq2, except uses a different statistical method (similar to Limma Voom).)</p></li>
<li><p>Parameter estimation and estimation of variance using the general linear model.</p></li>
<li><p>Identification of:</p>
<ul>
<li><strong>Coefficients:</strong> indicating overall expression strength</li>
<li><strong>Beta values:</strong> estimates of fold changes</li>
</ul></li>
</ul>
<p><strong>Step 3:</strong> Test for significant differences between conditions</p>
<p>After performing all analysis steps, we will explore the sample QC plots and plotting of results. In addition, we will use the html interface available through the sleuth package.</p>
</section>
<section id="sleuth-workflow" class="level2">
<h2 class="anchored" data-anchor-id="sleuth-workflow">Sleuth workflow</h2>
<section id="step-1-create-sleuth-object-for-analysis" class="level3">
<h3 class="anchored" data-anchor-id="step-1-create-sleuth-object-for-analysis">Step 1: Create Sleuth object for analysis</h3>
<p align="center">
<img src="../img/sleuth_workflow1.png" width="600">
</p>
<p>Similar to DESeq2, we need to tell Sleuth where to find the <strong>metadata</strong> (specifying which samplegroups the samples belong to, and any other metadata we want included in the analysis), <strong>estimated counts</strong> (output from Salmon) and the <strong>design formula</strong>. In addition, we also need <strong>annotables</strong> to easily convert between transcript IDs and associated gene names.</p>
<p>To create this Sleuth object, we need to perform the following steps:</p>
<ol type="1">
<li><p>Create a dataframe containing metadata and locations of the estimated counts files:</p>
<ul>
<li>including any columns containing metadata to used in the analysis</li>
<li>a column named <code>sample</code> containing all of the sample names matching the names in the metadata file</li>
<li>a column named <code>path</code> containing the path to the abundance estimate files output from <code>wasabi</code></li>
</ul></li>
<li><p>Create a variable containing the model design</p></li>
<li><p>Use annotables to create a dataset for Sleuth to query for Ensembl IDs and associated gene names</p></li>
</ol>
<section id="create-a-dataframe-needed-to-generate-sleuth-analysis-object" class="level4">
<h4 class="anchored" data-anchor-id="create-a-dataframe-needed-to-generate-sleuth-analysis-object">Create a dataframe needed to generate Sleuth analysis object</h4>
<p>First we need to bring in the metadata file:</p>
<ul>
<li>Create <strong>metadata</strong> associated with the Salmon files as we did in the DESeq2 lesson using the <code>data.frame()</code> function to ensure it is a dataframe.</li>
<li>Then combine the metadata with the paths to the transcript abundance files to use as input for the Sleuth analysis.</li>
</ul>
<div class="sourceCode" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-2"><a href="#cb4-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sample names from sfdirs</span></span>
<span id="cb4-3"><a href="#cb4-3" aria-hidden="true" tabindex="-1"></a>sf_dirs_samples <span class="ot">&lt;-</span> sf_dirs <span class="sc">%&gt;%</span></span>
<span id="cb4-4"><a href="#cb4-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">basename</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb4-5"><a href="#cb4-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.salmon"</span>, <span class="st">""</span>)</span>
<span id="cb4-6"><a href="#cb4-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb4-7"><a href="#cb4-7" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a sampletable/metadata</span></span>
<span id="cb4-8"><a href="#cb4-8" aria-hidden="true" tabindex="-1"></a>sampletype <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"MOV10_knockdown"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"MOV10_overexpression"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"control"</span>,<span class="dv">3</span>) ))</span>
<span id="cb4-9"><a href="#cb4-9" aria-hidden="true" tabindex="-1"></a>summarydata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sampletype, <span class="at">row.names =</span> sf_dirs_samples)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>We make sure the metadata and count estimate sample names match:</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb5-1"><a href="#cb5-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure the order of the `sfdirs` created above matches the order of samples in the `summarydata` rownames</span></span>
<span id="cb5-2"><a href="#cb5-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb5-3"><a href="#cb5-3" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sf_dirs_samples <span class="sc">==</span> <span class="fu">rownames</span>(summarydata))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Now, we can name the vector of directory paths with the corresponding sample names</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the directory paths for the abundance files with their corresponding sample IDs</span></span>
<span id="cb6-2"><a href="#cb6-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sf_dirs) <span class="ot">&lt;-</span> sf_dirs_samples</span>
<span id="cb6-4"><a href="#cb6-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" aria-hidden="true" tabindex="-1"></a>sf_dirs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>`</p>
<p>Sleuth expects the data to be presented in a specific format with specific column and row names; therefore, we will need to add columns to <code>summarydata</code> and name columns based on the sleuth requirements for the analysis.</p>
<p>Sleuth requires a column entitled “sample” containing the sample names:</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a column named 'sample'</span></span>
<span id="cb7-2"><a href="#cb7-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" aria-hidden="true" tabindex="-1"></a>summarydata<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(summarydata)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Now, we can include the path to the count estimate folders. Sleuth requires a column entitled “path” containing the paths to the estimated counts files stored in our <code>sf_dirs</code>:</p>
<div class="sourceCode" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a column named 'path'</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a>summarydata<span class="sc">$</span>path <span class="ot">&lt;-</span> sf_dirs</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
<section id="provide-the-model-design" class="level4">
<h4 class="anchored" data-anchor-id="provide-the-model-design">Provide the model design</h4>
<p>Now that we have the metadata and location of the count estimates, we can input our design formula to determine the covariates and/or confounders that should be included in your experimental design model. Sleuth can be used to analyze multiple conditions from complex experimental designs.</p>
<p>Within Sleuth, models are written similar to DESeq2. Since the only condition we plan to test is our sample type, our design formula is very simple:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb9-1"><a href="#cb9-1" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="er">~</span> sampletype</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>More complex designs can be analyzed using Sleuth as well by adding additional covariates (i.e <code>design &lt;- ~ sex + treatment</code>). Interaction terms can also be added to the design formula to test if the effect attributable to a given condition is different based on another factor, for example, if the treatment effect differs between sexes. To learn more about setting up design formulas for more complex designs, the <a href="http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html">DESeq2 vignette</a> has detailed documentation that should work for sleuth too.</p>
<p>In addition, <a href="https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html">one of the sleuth walk-throughs</a> describes in detail how to perform the LRT test with specification of the full and reduced models.</p>
</section>
<section id="query-annotables-dataset-to-obtain-the-corresponding-ensembl-transcriptgene-ids" class="level4">
<h4 class="anchored" data-anchor-id="query-annotables-dataset-to-obtain-the-corresponding-ensembl-transcriptgene-ids">Query annotables dataset to obtain the corresponding Ensembl transcript/gene IDs</h4>
<p>The last component to include for our analysis is the annotables Ensembl genome dataset to obtain the Ensembl transcript/gene IDs and gene names for annotation of results. There are conversion objects available to us by just loading the annotables library.</p>
<p>We can use the gene symbols from <code>grch38</code> and join this information with the <code>grch38_tx2gene</code> dataset for our conversion purposes. Again, we will need to provide column names that are consistent with what Sleuth is expecting.</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Inner join the tx2gene so we get gene symbols</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a>t2g <span class="ot">&lt;-</span> grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(symbol, ensgene) <span class="sc">%&gt;%</span> </span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(grch38_tx2gene) <span class="sc">%&gt;%</span> </span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">target_id =</span> enstxp, </span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a>         <span class="at">ens_gene =</span> ensgene, </span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a>         <span class="at">ext_gene =</span> symbol)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</section>
</section>
<section id="step-2-fit-the-sleuth-model" class="level3">
<h3 class="anchored" data-anchor-id="step-2-fit-the-sleuth-model">Step 2: Fit the sleuth model</h3>
<p align="center">
<img src="../img/sleuth_workflow2_updated.png" width="600">
</p>
<section id="fit-the-transcript-abundance-data-to-the-sleuth-model" class="level4">
<h4 class="anchored" data-anchor-id="fit-the-transcript-abundance-data-to-the-sleuth-model">Fit the transcript abundance data to the Sleuth model</h4>
<p>Using the <code>sleuth_prep()</code> function, the counts are normalized and filtered, then merged with the metadata. In addition, the bootstraps for each transcript are summarized. This function can take a bit of time, but there is an option (<code>ncores</code>) to split across multiple processors.</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb11-1"><a href="#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create sleuth object for analysis </span></span>
<span id="cb11-2"><a href="#cb11-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb11-3"><a href="#cb11-3" aria-hidden="true" tabindex="-1"></a>so <span class="ot">&lt;-</span> <span class="fu">sleuth_prep</span>(summarydata, </span>
<span id="cb11-4"><a href="#cb11-4" aria-hidden="true" tabindex="-1"></a>                  <span class="at">full_model =</span> design, </span>
<span id="cb11-5"><a href="#cb11-5" aria-hidden="true" tabindex="-1"></a>                  <span class="at">target_mapping =</span> t2g, </span>
<span id="cb11-6"><a href="#cb11-6" aria-hidden="true" tabindex="-1"></a>                  <span class="at">read_bootstrap_tpm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-7"><a href="#cb11-7" aria-hidden="true" tabindex="-1"></a>                  <span class="at">extra_bootstrap_summary =</span> <span class="cn">TRUE</span>,</span>
<span id="cb11-8"><a href="#cb11-8" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transformation_function =</span> <span class="cf">function</span>(x) <span class="fu">log2</span>(x <span class="sc">+</span> <span class="fl">0.5</span>))  </span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<blockquote class="blockquote">
<p><strong>NOTE:</strong> By default the transformation of counts is natural log, which would make the output fold changes somewhat more difficult to interpret. By specifying the <code>transformation_function</code> to be <code>log2(x + 0.5)</code> we are ensuring our output fold changes are log2.</p>
</blockquote>
<p>In fitting the sleuth model, sleuth performs shrinkage of variance, parameter estimation and estimation of variance using the general linear model:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a>so <span class="ot">&lt;-</span> <span class="fu">sleuth_fit</span>(so)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<blockquote class="blockquote">
<p><strong>NOTE:</strong> alternatively the two prior steps could have been run as:</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb13-1"><a href="#cb13-1" aria-hidden="true" tabindex="-1"></a>so <span class="ot">&lt;-</span> <span class="fu">sleuth_prep</span>(summarydata, </span>
<span id="cb13-2"><a href="#cb13-2" aria-hidden="true" tabindex="-1"></a>                 <span class="at">full_model =</span> design, </span>
<span id="cb13-3"><a href="#cb13-3" aria-hidden="true" tabindex="-1"></a>                 <span class="at">target_mapping =</span> t2g, </span>
<span id="cb13-4"><a href="#cb13-4" aria-hidden="true" tabindex="-1"></a>                 <span class="at">read_bootstrap_tpm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-5"><a href="#cb13-5" aria-hidden="true" tabindex="-1"></a>                 <span class="at">extra_bootstrap_summary =</span> <span class="cn">TRUE</span>,</span>
<span id="cb13-6"><a href="#cb13-6" aria-hidden="true" tabindex="-1"></a>                 <span class="at">transformation_function =</span> <span class="cf">function</span>(x) <span class="fu">log2</span>(x <span class="sc">+</span> <span class="fl">0.5</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb13-7"><a href="#cb13-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">sleuth_fit</span>()</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</blockquote>
</section>
<section id="check-which-models-have-been-fit-and-which-coefficients-can-be-tested" class="level4">
<h4 class="anchored" data-anchor-id="check-which-models-have-been-fit-and-which-coefficients-can-be-tested">Check which models have been fit and which coefficients can be tested</h4>
<p>Ensure the design model and coefficients are correct for your analysis. The level not shown is the base level.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">models</span>(so)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<blockquote class="blockquote">
<p><strong>NOTE:</strong> Sleuth will automatically use the first level (alphabetically by default) in the factor variable being tested to compare all other conditions against (in our metadata, this is ‘control’). If you want to use a different condition to be the base level, then you would need to use the relevel() function to change the base level of the variable in step 1 above. For example, if we wanted the base level of <code>sampletype</code> to be “MOV10_knockdown”, we could use the following code:</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb15-1"><a href="#cb15-1" aria-hidden="true" tabindex="-1"></a><span class="co"># DO NOT RUN!</span></span>
<span id="cb15-2"><a href="#cb15-2" aria-hidden="true" tabindex="-1"></a>summarydata<span class="sc">$</span>sampletype <span class="ot">&lt;-</span> <span class="fu">relevel</span>(summarydata<span class="sc">$</span>sampletype, <span class="at">ref =</span> <span class="st">"MOV10_knockdown"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p><strong><em>An ordered factor will not give interpretable output, so do NOT order the factor using the factor() function with <code>ordered = TRUE</code>, use factor() specifying the levels or relevel() instead.</em></strong></p>
</blockquote>
</section>
</section>
<section id="step-3-test-significant-differences-between-conditions-using-the-wald-test" class="level3">
<h3 class="anchored" data-anchor-id="step-3-test-significant-differences-between-conditions-using-the-wald-test">Step 3: Test significant differences between conditions using the Wald test</h3>
<p>At this step in the workflow, we need to specify which level we want to compare against the base level (use the name given for the coefficients from <code>models(so)</code>):</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Wald test for differential expression of isoforms</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a>oe <span class="ot">&lt;-</span> <span class="fu">sleuth_wt</span>(so, </span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a>                <span class="at">which_beta =</span> <span class="st">'sampletypeMOV10_overexpression'</span>)</span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-6"><a href="#cb16-6" aria-hidden="true" tabindex="-1"></a><span class="co"># output results</span></span>
<span id="cb16-7"><a href="#cb16-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-8"><a href="#cb16-8" aria-hidden="true" tabindex="-1"></a>sleuth_results_oe <span class="ot">&lt;-</span> <span class="fu">sleuth_results</span>(oe, </span>
<span id="cb16-9"><a href="#cb16-9" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">test =</span> <span class="st">'sampletypeMOV10_overexpression'</span>, </span>
<span id="cb16-10"><a href="#cb16-10" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">show_all =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<blockquote class="blockquote">
<p><strong>NOTE:</strong></p>
</blockquote>
<p>The output represents the results from the differential expression testing with the following columns:</p>
<ul>
<li><p><strong><code>target_id</code>:</strong> the Ensembl transcript ID</p></li>
<li><p><strong><code>pval</code>:</strong> the Wald test FDR adjusted pvalue using Benjamini-Hochberg</p></li>
<li><p><strong><code>qval</code>:</strong> the p-value adjusted for multiple test correction</p></li>
<li><p><strong><code>b</code>:</strong> beta value, which is the log2 fold changes between conditions (These are log2 b/c we specified log2 transformation in the <code>sleuth_prep()</code> step. By default, these would have been natural log fold changes).</p></li>
<li><p><strong><code>se_b</code>:</strong> standard error of the beta value</p></li>
<li><p><strong><code>mean_obs</code>:</strong> the mean expression (log2) of the transcript across all samples</p></li>
<li><p><strong><code>var_obs</code>:</strong> the biological variance of the expression</p></li>
<li><p><strong><code>tech_var</code>:</strong> the technical variance of expression (derived from the bootstraps)</p></li>
<li><p><strong><code>sigma_sq</code>:</strong> raw estimator of the variance once the technical variance has been removed</p></li>
<li><p><strong><code>smooth_sigma_sq</code>:</strong> the smooth regression fit for the shrinkage estimation</p></li>
<li><p><strong><code>final_sigma_sq</code>:</strong> max(sigma_sq, smooth_sigma_sq). this is the one used for covariance estimation of beta (in addition to tech_var)</p></li>
<li><p><strong><code>ens_gene</code>:</strong> associated Ensembl gene ID</p></li>
<li><p><strong><code>ext_gene</code>:</strong> associated gene symbol</p></li>
<li><p>Adapted from the results table description in Harold Pimental’s <a href="https://groups.google.com/forum/#!searchin/kallisto-sleuth-users/mean_obs%7Csort:date/kallisto-sleuth-users/kWodd7CQejE/nzr6hEOKAAAJ">post</a> to the ‘kallisto-sleuth-users’ google groups. Harold Pimental is an author of sleuth.*</p></li>
</ul>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../img/sleuth_mov10_results.png" class="img-fluid figure-img"></p>
<figcaption>sleuth_results</figcaption>
</figure>
</div>
</section>
<section id="exploring-transcript-level-expression-between-samples" class="level3">
<h3 class="anchored" data-anchor-id="exploring-transcript-level-expression-between-samples">Exploring transcript-level expression between samples</h3>
<section id="exploratory-analyses" class="level4">
<h4 class="anchored" data-anchor-id="exploratory-analyses">Exploratory analyses:</h4>
<p>Now that we have our results, we can perform some exploratory analyses, such as PCA, heatmap, and distributions of counts between conditions. By default, these plots will use the sleuth-normalized <code>est_counts</code> (not the log2 transformed values).</p>
<p><strong>PCA:</strong> There are multiple functions to explore the variation in the dataset explained by the different PCs.</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca</span>(oe, </span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>         <span class="at">color_by =</span> <span class="st">'sampletype'</span>,</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a>         <span class="at">text_labels =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/pca_sleuth.png" width="700">
</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb18-1"><a href="#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pc_variance</span>(oe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/sleuth_pca_variance.png" width="500">
</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_loadings</span>(oe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/sleuth_pca_loadings.png" width="500">
</p>
<p><strong>Heatmap:</strong> The heatmap plot is shaded with the Jensen-Shannon divergence values. Therefore, <strong>lower divergence values represent samples that are more similar to each other</strong>.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb20-1"><a href="#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_sample_heatmap</span>(oe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/sleuth_cor_heatmap.png" width="600">
</p>
<p><strong>Count distributions:</strong> There is a histogram plot to explore count distributions between sample groups, which should be similar to each other when performing DE testing. The count distributions represent the proportion of genes (on the y-axis) associated with the number of counts (designated on the x-axis):</p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_group_density</span>(oe, </span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_filtered =</span> <span class="cn">FALSE</span>, </span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">units =</span> <span class="st">"est_counts"</span>,</span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trans =</span> <span class="st">"log"</span>, </span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">grouping =</span> <span class="st">"sampletype"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/plot_density_unfiltered.png" width="600">
</p>
<p>As we know, most genes have few counts, but we filter these genes prior to performing DE analysis. If we want to look at the distributions of the filtered genes used for DE analysis, we could change the <code>use_filtered</code> argument to <code>TRUE</code>.</p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb22-1"><a href="#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_group_density</span>(oe, </span>
<span id="cb22-2"><a href="#cb22-2" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_filtered =</span> <span class="cn">TRUE</span>, </span>
<span id="cb22-3"><a href="#cb22-3" aria-hidden="true" tabindex="-1"></a>                   <span class="at">units =</span> <span class="st">"est_counts"</span>,</span>
<span id="cb22-4"><a href="#cb22-4" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trans =</span> <span class="st">"log"</span>, </span>
<span id="cb22-5"><a href="#cb22-5" aria-hidden="true" tabindex="-1"></a>                   <span class="at">grouping =</span> <span class="st">"sampletype"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/plot_density_filtered.png" width="600">
</p>
</section>
<section id="results-analyses" class="level4">
<h4 class="anchored" data-anchor-id="results-analyses">Results analyses:</h4>
<p>There are also functions to explore the results. For example, we can perform an expression heatmap for select transcripts:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a>sig_transcripts <span class="ot">&lt;-</span> sleuth_results_oe <span class="sc">%&gt;%</span> </span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(qval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_transcript_heatmap</span>(oe, </span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a>                        <span class="at">transcripts =</span> sig_transcripts<span class="sc">$</span>target_id[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>])</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/heatmap_sleuth.png" width="600">
</p>
<p>Sleuth also has some handy functions to plot expression of transcripts with bootstrap variation to <strong>visualize both biological and technical variation</strong> for selected transcripts:</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb24-1"><a href="#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting</span></span>
<span id="cb24-2"><a href="#cb24-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-3"><a href="#cb24-3" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bootstrap</span>(oe, </span>
<span id="cb24-4"><a href="#cb24-4" aria-hidden="true" tabindex="-1"></a>               <span class="at">target_id =</span> <span class="st">"ENST00000495374"</span>, </span>
<span id="cb24-5"><a href="#cb24-5" aria-hidden="true" tabindex="-1"></a>               <span class="at">units =</span> <span class="st">"est_counts"</span>, </span>
<span id="cb24-6"><a href="#cb24-6" aria-hidden="true" tabindex="-1"></a>               <span class="at">color_by =</span> <span class="st">"sampletype"</span>)</span>
<span id="cb24-7"><a href="#cb24-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb24-8"><a href="#cb24-8" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bootstrap</span>(oe, </span>
<span id="cb24-9"><a href="#cb24-9" aria-hidden="true" tabindex="-1"></a>               <span class="at">target_id =</span> <span class="st">"ENST00000367412"</span>, </span>
<span id="cb24-10"><a href="#cb24-10" aria-hidden="true" tabindex="-1"></a>               <span class="at">units =</span> <span class="st">"est_counts"</span>, </span>
<span id="cb24-11"><a href="#cb24-11" aria-hidden="true" tabindex="-1"></a>               <span class="at">color_by =</span> <span class="st">"sampletype"</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p align="center">
<img src="../img/sleuth_bootstraps1.png" width="600">
</p>
<p align="center">
<img src="../img/sleuth_bootstraps2.png" width="600">
</p>
<p>Sleuth also offers us the option to explore the data and results interactively using a web interface.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sleuth_live</span>(oe)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>Using this web interface, we can explore diagnostic plots and summaries of experimental factors and expression levels. We also have the ability to perform clustering analyses such as PCA and heatmaps. Finally, we can analyze the differential expression results by plotting MA and volcano plots and by exploring expression levels at the transcript and gene levels. A brief tutorial explaining the options available interactively can be found <a href="http://www.rna-seqblog.com/iihg-intro-to-sleuth-for-rna-seq/">here</a>.</p>
<p>Explore the results table under the <code>analyses</code> tab, <code>test table</code>.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../img/mov10_test_table.png" class="img-fluid figure-img"></p>
<figcaption>results_table</figcaption>
</figure>
</div>
<p>Look at the expression levels of Mov10 for three different isoforms using the <code>analyses</code> tab, <code>gene view</code>. Look for the Mov10 gene, ENSG00000155363. Compare expression of the different isoforms of Mov10.</p>
<div class="quarto-figure quarto-figure-center">
<figure class="figure">
<p><img src="../img/mov10_isoform_expression.png" class="img-fluid figure-img"></p>
<figcaption>mov_isoforms</figcaption>
</figure>
</div>
<hr>
<p><em>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a>. These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</em></p>


<!-- -->

</section>
</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  const viewSource = window.document.getElementById('quarto-view-source') ||
                     window.document.getElementById('quarto-code-tools-source');
  if (viewSource) {
    const sourceUrl = viewSource.getAttribute("data-quarto-source-url");
    viewSource.addEventListener("click", function(e) {
      if (sourceUrl) {
        // rstudio viewer pane
        if (/\bcapabilities=\b/.test(window.location)) {
          window.open(sourceUrl);
        } else {
          window.location.href = sourceUrl;
        }
      } else {
        const modal = new bootstrap.Modal(document.getElementById('quarto-embedded-source-code-modal'));
        modal.show();
      }
      return false;
    });
  }
  function toggleCodeHandler(show) {
    return function(e) {
      const detailsSrc = window.document.querySelectorAll(".cell > details > .sourceCode");
      for (let i=0; i<detailsSrc.length; i++) {
        const details = detailsSrc[i].parentElement;
        if (show) {
          details.open = true;
        } else {
          details.removeAttribute("open");
        }
      }
      const cellCodeDivs = window.document.querySelectorAll(".cell > .sourceCode");
      const fromCls = show ? "hidden" : "unhidden";
      const toCls = show ? "unhidden" : "hidden";
      for (let i=0; i<cellCodeDivs.length; i++) {
        const codeDiv = cellCodeDivs[i];
        if (codeDiv.classList.contains(fromCls)) {
          codeDiv.classList.remove(fromCls);
          codeDiv.classList.add(toCls);
        } 
      }
      return false;
    }
  }
  const hideAllCode = window.document.getElementById("quarto-hide-all-code");
  if (hideAllCode) {
    hideAllCode.addEventListener("click", toggleCodeHandler(false));
  }
  const showAllCode = window.document.getElementById("quarto-show-all-code");
  if (showAllCode) {
    showAllCode.addEventListener("click", toggleCodeHandler(true));
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script><div class="modal fade" id="quarto-embedded-source-code-modal" tabindex="-1" aria-labelledby="quarto-embedded-source-code-modal-label" aria-hidden="true"><div class="modal-dialog modal-dialog-scrollable"><div class="modal-content"><div class="modal-header"><h5 class="modal-title" id="quarto-embedded-source-code-modal-label">Source Code</h5><button class="btn-close" data-bs-dismiss="modal"></button></div><div class="modal-body"><div class="">
<div class="sourceCode" id="cb26" data-shortcodes="false"><pre class="sourceCode markdown code-with-copy"><code class="sourceCode markdown"><span id="cb26-1"><a href="#cb26-1" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb26-2"><a href="#cb26-2" aria-hidden="true" tabindex="-1"></a><span class="an">title:</span><span class="co"> "Differential expression of transcripts using Sleuth"</span></span>
<span id="cb26-3"><a href="#cb26-3" aria-hidden="true" tabindex="-1"></a><span class="an">author:</span><span class="co"> "Mary Piper"</span></span>
<span id="cb26-4"><a href="#cb26-4" aria-hidden="true" tabindex="-1"></a><span class="an">date:</span><span class="co"> "Tuesday, April 10th, 2018"</span></span>
<span id="cb26-5"><a href="#cb26-5" aria-hidden="true" tabindex="-1"></a><span class="co">---</span></span>
<span id="cb26-6"><a href="#cb26-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-7"><a href="#cb26-7" aria-hidden="true" tabindex="-1"></a>Approximate time: 1.5 hours</span>
<span id="cb26-8"><a href="#cb26-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-9"><a href="#cb26-9" aria-hidden="true" tabindex="-1"></a><span class="fu">## Learning Objectives</span></span>
<span id="cb26-10"><a href="#cb26-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-11"><a href="#cb26-11" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Determine differential expression of isoforms and visualization of results using Sleuth</span>
<span id="cb26-12"><a href="#cb26-12" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Understand how Sleuth determines biological and technical variation</span>
<span id="cb26-13"><a href="#cb26-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-14"><a href="#cb26-14" aria-hidden="true" tabindex="-1"></a><span class="fu">## Transcript-level differential expression</span></span>
<span id="cb26-15"><a href="#cb26-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-16"><a href="#cb26-16" aria-hidden="true" tabindex="-1"></a>Until this point we have focused on looking for expression changes at the gene-level. However, if you are interested in looking at **splice isoform expression changes between groups** the previous methods (i.e DESeq2) will not work. To demonstrate how to identify transcript-level differential expression we will be using a tool called Sleuth.</span>
<span id="cb26-17"><a href="#cb26-17" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-18"><a href="#cb26-18" aria-hidden="true" tabindex="-1"></a>&lt;img src="../img/workflow-salmon-sleuth.png" width="700"&gt;</span>
<span id="cb26-19"><a href="#cb26-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-20"><a href="#cb26-20" aria-hidden="true" tabindex="-1"></a><span class="fu">## What is Sleuth?</span></span>
<span id="cb26-21"><a href="#cb26-21" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-22"><a href="#cb26-22" aria-hidden="true" tabindex="-1"></a><span class="co">[</span><span class="ot">Sleuth</span><span class="co">](http://pachterlab.github.io/sleuth/)</span> is a fast, lightweight tool that uses transcript abundance estimates output from **pseudo-alignment** algorithms that use **bootstrap sampling**, such as Sailfish, Salmon, and Kallisto, to perform differential expression analysis of gene isoforms. </span>
<span id="cb26-23"><a href="#cb26-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-24"><a href="#cb26-24" aria-hidden="true" tabindex="-1"></a>To analyze the differential expression of gene isoforms, it is expected that RNA-Seq reads will often align to multiple isoforms of the same gene. Therefore, **multimapping reads cannot be ignored** to properly determine abundances of gene isoforms. </span>
<span id="cb26-25"><a href="#cb26-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-26"><a href="#cb26-26" aria-hidden="true" tabindex="-1"></a>Due to the statistical procedure required to assign reads to gene isoforms, in addition to the random processes underlying RNA-Seq, there will be **technical variability in the abundance estimates** output from the pseudo-alignment tool [[2](https://rawgit.com/pachterlab/sleuth/master/inst/doc/intro.html), [3](https://www.nature.com/articles/nmeth.4324)]. Therefore, **we would need technical replicates to distinguish technical variability from the biological variability**.</span>
<span id="cb26-27"><a href="#cb26-27" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-28"><a href="#cb26-28" aria-hidden="true" tabindex="-1"></a>Sleuth accounts for this technical variability by using **bootstraps as a proxy for technical replicates**, which are used to model the technical variability in the abundance estimates. Bootstrapping essentially **calculates the abundance estimates for all genes using a different sub-sample of reads** during each round of bootstrapping. The variation in the abundance estimates output from each round of bootstrapping is used for the estimation of the technical variance for each gene. </span>
<span id="cb26-29"><a href="#cb26-29" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-30"><a href="#cb26-30" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-31"><a href="#cb26-31" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_tech_var.png" width="600"/&gt;</span>
<span id="cb26-32"><a href="#cb26-32" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-33"><a href="#cb26-33" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-34"><a href="#cb26-34" aria-hidden="true" tabindex="-1"></a>_Adapted from: Nature Methods **14**, 687–690 (2017)_</span>
<span id="cb26-35"><a href="#cb26-35" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-36"><a href="#cb26-36" aria-hidden="true" tabindex="-1"></a>Sleuth models the unobserved true abundance (logarithm of true counts) using a general linear model, but includes the technical variance (variance between bootstrapping runs) as error in the response variable. </span>
<span id="cb26-37"><a href="#cb26-37" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-38"><a href="#cb26-38" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-39"><a href="#cb26-39" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_formula1.png" width="500"/&gt;</span>
<span id="cb26-40"><a href="#cb26-40" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-41"><a href="#cb26-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-42"><a href="#cb26-42" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-43"><a href="#cb26-43" aria-hidden="true" tabindex="-1"></a>The observed (log) abundance estimates represent the sum of the true counts and the technical noise. Thereby, sleuth teases apart the source of the variance (technical vs. biological) for estimation of the "true" biological variance when determining whether transcripts are differentially expressed.</span>
<span id="cb26-44"><a href="#cb26-44" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-45"><a href="#cb26-45" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-46"><a href="#cb26-46" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_formula2.png" width="425"/&gt;</span>
<span id="cb26-47"><a href="#cb26-47" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-48"><a href="#cb26-48" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-49"><a href="#cb26-49" aria-hidden="true" tabindex="-1"></a>In addition to performing differential expression analysis of transcripts, the sleuth tool also provides an HTML interface allowing exploration of the data and differential expression results interactively. More information about the theory/process for sleuth is available in the <span class="co">[</span><span class="ot">Nature Methods paper</span><span class="co">](https://www.nature.com/articles/nmeth.4324)</span>, this <span class="co">[</span><span class="ot">blogpost</span><span class="co">](https://liorpachter.wordpress.com/2015/08/17/a-sleuth-for-rna-seq/)</span> and step-by-step tutorials are available on the <span class="co">[</span><span class="ot">sleuth website</span><span class="co">](https://pachterlab.github.io/sleuth/walkthroughs)</span>.</span>
<span id="cb26-50"><a href="#cb26-50" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-51"><a href="#cb26-51" aria-hidden="true" tabindex="-1"></a>***NOTE:*** *Kallisto is distributed under a non-commercial license, while Sailfish and Salmon are distributed under the [GNU General Public License, version 3](http://www.gnu.org/licenses/gpl.html).*</span>
<span id="cb26-52"><a href="#cb26-52" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-53"><a href="#cb26-53" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-54"><a href="#cb26-54" aria-hidden="true" tabindex="-1"></a><span class="fu">## Set-up for Running Sleuth</span></span>
<span id="cb26-55"><a href="#cb26-55" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-56"><a href="#cb26-56" aria-hidden="true" tabindex="-1"></a>Sleuth is a lightweight algorithm that can be quickly run on our personal computers. Let's **open a new project called `DE_sleuth`**.</span>
<span id="cb26-57"><a href="#cb26-57" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-58"><a href="#cb26-58" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; **NOTE:** We could have used the previous Salmon project, however we did not include the </span><span class="in">`--numBootstraps 30`</span><span class="at"> parameter in our automation script and so we do not have the appropriate files.</span></span>
<span id="cb26-59"><a href="#cb26-59" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-60"><a href="#cb26-60" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-61"><a href="#cb26-61" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Download the Salmon files with bootstraps <span class="co">[</span><span class="ot">from here</span><span class="co">](https://www.dropbox.com/s/ntpkgkxikb46b35/data.zip?dl=1)</span>. Move it into your working directory and decompress the file. You should now see a <span class="in">`data`</span> directory.</span>
<span id="cb26-62"><a href="#cb26-62" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Create a new R script ('File' -&gt; 'New File' -&gt; 'Rscript'), and save it as <span class="in">`sleuth_de.R`</span></span>
<span id="cb26-63"><a href="#cb26-63" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-64"><a href="#cb26-64" aria-hidden="true" tabindex="-1"></a>To perform any analysis, we need to load the libraries for <span class="in">`wasabi`</span> and <span class="in">`sleuth`</span>. Sleuth also requires <span class="in">`annotables`</span>, so this package will be loaded as well:</span>
<span id="cb26-65"><a href="#cb26-65" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-66"><a href="#cb26-66" aria-hidden="true" tabindex="-1"></a><span class="in">```R</span></span>
<span id="cb26-67"><a href="#cb26-67" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(wasabi)</span>
<span id="cb26-68"><a href="#cb26-68" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sleuth)</span>
<span id="cb26-69"><a href="#cb26-69" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(annotables)</span>
<span id="cb26-70"><a href="#cb26-70" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyverse)</span>
<span id="cb26-71"><a href="#cb26-71" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-72"><a href="#cb26-72" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-73"><a href="#cb26-73" aria-hidden="true" tabindex="-1"></a><span class="fu">## Using Wasabi to convert Salmon output for Sleuth</span></span>
<span id="cb26-74"><a href="#cb26-74" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-75"><a href="#cb26-75" aria-hidden="true" tabindex="-1"></a>Sleuth was built to use the bootstrapped estimates of transcript abundance from Kallisto; however, abundance estimates from Salmon (or Sailfish) work just as well, as long as bootstrapping is performed. To use our Salmon output estimates, we need to convert them to a Sleuth-compatible format using the Wasabi package.</span>
<span id="cb26-76"><a href="#cb26-76" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-77"><a href="#cb26-77" aria-hidden="true" tabindex="-1"></a>First, we create a simple vector containing the paths to the directories containing the transcript abundance estimates for each sample (folders containing the .quant files). We can use the <span class="in">`file.path()`</span> function to give the paths to each of the directories. </span>
<span id="cb26-78"><a href="#cb26-78" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-79"><a href="#cb26-79" aria-hidden="true" tabindex="-1"></a>Now, let's use this function to create our list of the paths to our transcript abundance files:</span>
<span id="cb26-80"><a href="#cb26-80" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-81"><a href="#cb26-81" aria-hidden="true" tabindex="-1"></a><span class="in">```R</span></span>
<span id="cb26-82"><a href="#cb26-82" aria-hidden="true" tabindex="-1"></a>sf_dirs <span class="ot">&lt;-</span> <span class="fu">file.path</span>(<span class="st">"data"</span>, <span class="fu">c</span>(<span class="st">"Mov10_kd_2.salmon"</span>, <span class="st">"Mov10_kd_3.salmon"</span>, <span class="st">"Mov10_oe_1.salmon"</span>, <span class="st">"Mov10_oe_2.salmon"</span>, <span class="st">"Mov10_oe_3.salmon"</span>,<span class="st">"Irrel_kd_1.salmon"</span>, <span class="st">"Irrel_kd_2.salmon"</span>, <span class="st">"Irrel_kd_3.salmon"</span>))</span>
<span id="cb26-83"><a href="#cb26-83" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-84"><a href="#cb26-84" aria-hidden="true" tabindex="-1"></a>sf_dirs</span>
<span id="cb26-85"><a href="#cb26-85" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-86"><a href="#cb26-86" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-87"><a href="#cb26-87" aria-hidden="true" tabindex="-1"></a>Now, we simply run the <span class="in">`prepare_fish_for_sleuth`</span> function, which will write some status messages to the console and, when it's done, each directory will contain an <span class="in">`abundance.h5`</span> file in a sleuth-compatible format.</span>
<span id="cb26-88"><a href="#cb26-88" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-89"><a href="#cb26-89" aria-hidden="true" tabindex="-1"></a><span class="in">```R</span></span>
<span id="cb26-90"><a href="#cb26-90" aria-hidden="true" tabindex="-1"></a><span class="fu">prepare_fish_for_sleuth</span>(sf_dirs)</span>
<span id="cb26-91"><a href="#cb26-91" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-92"><a href="#cb26-92" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-93"><a href="#cb26-93" aria-hidden="true" tabindex="-1"></a>Each of the sample directories should now contain the <span class="in">`abundance.h5`</span> files. These </span>
<span id="cb26-94"><a href="#cb26-94" aria-hidden="true" tabindex="-1"></a>files will be used as input to Sleuth.</span>
<span id="cb26-95"><a href="#cb26-95" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-96"><a href="#cb26-96" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-97"><a href="#cb26-97" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_abund_h5.png" width="600"/&gt;</span>
<span id="cb26-98"><a href="#cb26-98" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-99"><a href="#cb26-99" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-100"><a href="#cb26-100" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sleuth for estimation of differential expression of transcripts</span></span>
<span id="cb26-101"><a href="#cb26-101" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-102"><a href="#cb26-102" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-103"><a href="#cb26-103" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_workflow.png" width="700"/&gt;</span>
<span id="cb26-104"><a href="#cb26-104" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-105"><a href="#cb26-105" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-106"><a href="#cb26-106" aria-hidden="true" tabindex="-1"></a>The workflow for Sleuth is similar to the workflow followed for DESeq2, even though, the models for estimating differential expression are very different. </span>
<span id="cb26-107"><a href="#cb26-107" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-108"><a href="#cb26-108" aria-hidden="true" tabindex="-1"></a>**Step 1:** Creation of Sleuth object to provide metadata, estimated counts, and design formula for the analysis, in addition to a <span class="in">`annotables`</span> database to switch between transcript IDs and associated gene names.</span>
<span id="cb26-109"><a href="#cb26-109" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-110"><a href="#cb26-110" aria-hidden="true" tabindex="-1"></a>**Step 2:** Fit the sleuth model</span>
<span id="cb26-111"><a href="#cb26-111" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-112"><a href="#cb26-112" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Estimation of size (normalization) factors using the median of ratios method (similar to DESeq2)</span>
<span id="cb26-113"><a href="#cb26-113" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-114"><a href="#cb26-114" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Normalization of estimated counts using size factors (est. counts / size factors - similar to DESeq2)</span>
<span id="cb26-115"><a href="#cb26-115" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-116"><a href="#cb26-116" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Filtering of low abundance transcripts (&lt; 5 est counts in more than 47% of the samples)</span>
<span id="cb26-117"><a href="#cb26-117" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-118"><a href="#cb26-118" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Normalization of technical variation estimates</span>
<span id="cb26-119"><a href="#cb26-119" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-120"><a href="#cb26-120" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Estimation of biological variance and shrinkage estimates (With small sample sizes, we will make very bad estimates of transcript-wise dispersion unless we share information across transcripts. Sleuth regularizes the biological variance estimate with shrinkage, similar to DESeq2, except uses a different statistical method (similar to Limma Voom).)</span>
<span id="cb26-121"><a href="#cb26-121" aria-hidden="true" tabindex="-1"></a>    </span>
<span id="cb26-122"><a href="#cb26-122" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Parameter estimation and estimation of variance using the general linear model.</span>
<span id="cb26-123"><a href="#cb26-123" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-124"><a href="#cb26-124" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Identification of:</span>
<span id="cb26-125"><a href="#cb26-125" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>**Coefficients:** indicating overall expression strength</span>
<span id="cb26-126"><a href="#cb26-126" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>**Beta values:** estimates of fold changes    </span>
<span id="cb26-127"><a href="#cb26-127" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-128"><a href="#cb26-128" aria-hidden="true" tabindex="-1"></a>**Step 3:** Test for significant differences between conditions</span>
<span id="cb26-129"><a href="#cb26-129" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-130"><a href="#cb26-130" aria-hidden="true" tabindex="-1"></a>After performing all analysis steps, we will explore the sample QC plots and plotting of results. In addition, we will use the html interface available through the sleuth package.</span>
<span id="cb26-131"><a href="#cb26-131" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-132"><a href="#cb26-132" aria-hidden="true" tabindex="-1"></a><span class="fu">## Sleuth workflow</span></span>
<span id="cb26-133"><a href="#cb26-133" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-134"><a href="#cb26-134" aria-hidden="true" tabindex="-1"></a><span class="fu">### Step 1: Create Sleuth object for analysis</span></span>
<span id="cb26-135"><a href="#cb26-135" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-136"><a href="#cb26-136" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-137"><a href="#cb26-137" aria-hidden="true" tabindex="-1"></a>&lt;img src="../img/sleuth_workflow1.png" width="600"/&gt;</span>
<span id="cb26-138"><a href="#cb26-138" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-139"><a href="#cb26-139" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-140"><a href="#cb26-140" aria-hidden="true" tabindex="-1"></a>Similar to DESeq2, we need to tell Sleuth where to find the **metadata** (specifying which samplegroups the samples belong to, and any other metadata we want included in the analysis), **estimated counts** (output from Salmon) and the **design formula**. In addition, we also need **annotables** to easily convert between transcript IDs and associated gene names. </span>
<span id="cb26-141"><a href="#cb26-141" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-142"><a href="#cb26-142" aria-hidden="true" tabindex="-1"></a>To create this Sleuth object, we need to perform the following steps:</span>
<span id="cb26-143"><a href="#cb26-143" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-144"><a href="#cb26-144" aria-hidden="true" tabindex="-1"></a><span class="ss">1. </span>Create a dataframe containing metadata and locations of the estimated counts files:</span>
<span id="cb26-145"><a href="#cb26-145" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-146"><a href="#cb26-146" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>including any columns containing metadata to used in the analysis</span>
<span id="cb26-147"><a href="#cb26-147" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>a column named <span class="in">`sample`</span> containing all of the sample names matching the names in the metadata file</span>
<span id="cb26-148"><a href="#cb26-148" aria-hidden="true" tabindex="-1"></a><span class="ss">    - </span>a column named <span class="in">`path`</span> containing the path to the abundance estimate files output from <span class="in">`wasabi`</span></span>
<span id="cb26-149"><a href="#cb26-149" aria-hidden="true" tabindex="-1"></a>        </span>
<span id="cb26-150"><a href="#cb26-150" aria-hidden="true" tabindex="-1"></a><span class="ss">2. </span>Create a variable containing the model design </span>
<span id="cb26-151"><a href="#cb26-151" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-152"><a href="#cb26-152" aria-hidden="true" tabindex="-1"></a><span class="ss">3. </span>Use annotables to create a dataset for Sleuth to query for Ensembl IDs and associated gene names</span>
<span id="cb26-153"><a href="#cb26-153" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-154"><a href="#cb26-154" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Create a dataframe needed to generate Sleuth analysis object</span></span>
<span id="cb26-155"><a href="#cb26-155" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-156"><a href="#cb26-156" aria-hidden="true" tabindex="-1"></a>First we need to bring in the metadata file:</span>
<span id="cb26-157"><a href="#cb26-157" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-158"><a href="#cb26-158" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Create **metadata** associated with the Salmon files as we did in the DESeq2 lesson using the <span class="in">`data.frame()`</span> function to ensure it is a dataframe. </span>
<span id="cb26-159"><a href="#cb26-159" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>Then combine the metadata with the paths to the transcript abundance files to use as input for the Sleuth analysis. </span>
<span id="cb26-160"><a href="#cb26-160" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-161"><a href="#cb26-161" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-162"><a href="#cb26-162" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-163"><a href="#cb26-163" aria-hidden="true" tabindex="-1"></a><span class="co"># Get sample names from sfdirs</span></span>
<span id="cb26-164"><a href="#cb26-164" aria-hidden="true" tabindex="-1"></a>sf_dirs_samples <span class="ot">&lt;-</span> sf_dirs <span class="sc">%&gt;%</span></span>
<span id="cb26-165"><a href="#cb26-165" aria-hidden="true" tabindex="-1"></a>  <span class="fu">basename</span>() <span class="sc">%&gt;%</span> </span>
<span id="cb26-166"><a href="#cb26-166" aria-hidden="true" tabindex="-1"></a>  <span class="fu">str_replace</span>(<span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.salmon"</span>, <span class="st">""</span>)</span>
<span id="cb26-167"><a href="#cb26-167" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-168"><a href="#cb26-168" aria-hidden="true" tabindex="-1"></a><span class="do">## Create a sampletable/metadata</span></span>
<span id="cb26-169"><a href="#cb26-169" aria-hidden="true" tabindex="-1"></a>sampletype <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">"MOV10_knockdown"</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">"MOV10_overexpression"</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">"control"</span>,<span class="dv">3</span>) ))</span>
<span id="cb26-170"><a href="#cb26-170" aria-hidden="true" tabindex="-1"></a>summarydata <span class="ot">&lt;-</span> <span class="fu">data.frame</span>(sampletype, <span class="at">row.names =</span> sf_dirs_samples)</span>
<span id="cb26-171"><a href="#cb26-171" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-172"><a href="#cb26-172" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-173"><a href="#cb26-173" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-174"><a href="#cb26-174" aria-hidden="true" tabindex="-1"></a>We make sure the metadata and count estimate sample names match:</span>
<span id="cb26-175"><a href="#cb26-175" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-176"><a href="#cb26-176" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-177"><a href="#cb26-177" aria-hidden="true" tabindex="-1"></a><span class="co"># Make sure the order of the `sfdirs` created above matches the order of samples in the `summarydata` rownames</span></span>
<span id="cb26-178"><a href="#cb26-178" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-179"><a href="#cb26-179" aria-hidden="true" tabindex="-1"></a><span class="fu">all</span>(sf_dirs_samples <span class="sc">==</span> <span class="fu">rownames</span>(summarydata))</span>
<span id="cb26-180"><a href="#cb26-180" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-181"><a href="#cb26-181" aria-hidden="true" tabindex="-1"></a>Now, we can name the vector of directory paths with the corresponding sample names</span>
<span id="cb26-182"><a href="#cb26-182" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-183"><a href="#cb26-183" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-184"><a href="#cb26-184" aria-hidden="true" tabindex="-1"></a><span class="co"># Name the directory paths for the abundance files with their corresponding sample IDs</span></span>
<span id="cb26-185"><a href="#cb26-185" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-186"><a href="#cb26-186" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(sf_dirs) <span class="ot">&lt;-</span> sf_dirs_samples</span>
<span id="cb26-187"><a href="#cb26-187" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-188"><a href="#cb26-188" aria-hidden="true" tabindex="-1"></a>sf_dirs</span>
<span id="cb26-189"><a href="#cb26-189" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-190"><a href="#cb26-190" aria-hidden="true" tabindex="-1"></a>`</span>
<span id="cb26-191"><a href="#cb26-191" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-192"><a href="#cb26-192" aria-hidden="true" tabindex="-1"></a>Sleuth expects the data to be presented in a specific format with specific column and row names; therefore, we will need to add columns to <span class="in">`summarydata`</span> and name columns based on the sleuth requirements for the analysis. </span>
<span id="cb26-193"><a href="#cb26-193" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-194"><a href="#cb26-194" aria-hidden="true" tabindex="-1"></a>Sleuth requires a column entitled "sample" containing the sample names:</span>
<span id="cb26-195"><a href="#cb26-195" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-196"><a href="#cb26-196" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-197"><a href="#cb26-197" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a column named 'sample'</span></span>
<span id="cb26-198"><a href="#cb26-198" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-199"><a href="#cb26-199" aria-hidden="true" tabindex="-1"></a>summarydata<span class="sc">$</span>sample <span class="ot">&lt;-</span> <span class="fu">rownames</span>(summarydata)</span>
<span id="cb26-200"><a href="#cb26-200" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-201"><a href="#cb26-201" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-202"><a href="#cb26-202" aria-hidden="true" tabindex="-1"></a>Now, we can include the path to the count estimate folders. Sleuth requires a column entitled "path" containing the paths to the estimated counts files stored in our <span class="in">`sf_dirs`</span>:</span>
<span id="cb26-203"><a href="#cb26-203" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-204"><a href="#cb26-204" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-205"><a href="#cb26-205" aria-hidden="true" tabindex="-1"></a><span class="co"># Adding a column named 'path'</span></span>
<span id="cb26-206"><a href="#cb26-206" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-207"><a href="#cb26-207" aria-hidden="true" tabindex="-1"></a>summarydata<span class="sc">$</span>path <span class="ot">&lt;-</span> sf_dirs</span>
<span id="cb26-208"><a href="#cb26-208" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-209"><a href="#cb26-209" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-210"><a href="#cb26-210" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-211"><a href="#cb26-211" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Provide the model design</span></span>
<span id="cb26-212"><a href="#cb26-212" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-213"><a href="#cb26-213" aria-hidden="true" tabindex="-1"></a>Now that we have the metadata and location of the count estimates, we can input our design formula to determine the covariates and/or confounders that should be included in your experimental design model. Sleuth can be used to analyze multiple conditions from complex experimental designs.</span>
<span id="cb26-214"><a href="#cb26-214" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-215"><a href="#cb26-215" aria-hidden="true" tabindex="-1"></a>Within Sleuth, models are written similar to DESeq2. Since the only condition we plan to test is our sample type, our design formula is very simple:</span>
<span id="cb26-216"><a href="#cb26-216" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-217"><a href="#cb26-217" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-218"><a href="#cb26-218" aria-hidden="true" tabindex="-1"></a>design <span class="ot">&lt;-</span> <span class="er">~</span> sampletype</span>
<span id="cb26-219"><a href="#cb26-219" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-220"><a href="#cb26-220" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-221"><a href="#cb26-221" aria-hidden="true" tabindex="-1"></a>More complex designs can be analyzed using Sleuth as well by adding additional covariates (i.e <span class="in">`design &lt;- ~ sex + treatment`</span>). Interaction terms can also be added to the design formula to test if the effect attributable to a given condition is different based on another factor, for example, if the treatment effect differs between sexes. To learn more about setting up design formulas for more complex designs, the <span class="co">[</span><span class="ot">DESeq2 vignette</span><span class="co">](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html)</span> has detailed documentation that should work for sleuth too. </span>
<span id="cb26-222"><a href="#cb26-222" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-223"><a href="#cb26-223" aria-hidden="true" tabindex="-1"></a>In addition, <span class="co">[</span><span class="ot">one of the sleuth walk-throughs</span><span class="co">](https://pachterlab.github.io/sleuth_walkthroughs/trapnell/analysis.html)</span> describes in detail how to perform the LRT test with specification of the full and reduced models.</span>
<span id="cb26-224"><a href="#cb26-224" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-225"><a href="#cb26-225" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Query annotables dataset to obtain the corresponding Ensembl transcript/gene IDs</span></span>
<span id="cb26-226"><a href="#cb26-226" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-227"><a href="#cb26-227" aria-hidden="true" tabindex="-1"></a>The last component to include for our analysis is the annotables Ensembl genome dataset to obtain the Ensembl transcript/gene IDs and gene names for annotation of results. There are conversion objects available to us by just loading the annotables library.</span>
<span id="cb26-228"><a href="#cb26-228" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-229"><a href="#cb26-229" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-230"><a href="#cb26-230" aria-hidden="true" tabindex="-1"></a>We can use the gene symbols from <span class="in">`grch38`</span> and join this information with the <span class="in">`grch38_tx2gene`</span> dataset for our conversion purposes. Again, we will need to provide column names that are consistent with what Sleuth is expecting.</span>
<span id="cb26-231"><a href="#cb26-231" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-232"><a href="#cb26-232" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-233"><a href="#cb26-233" aria-hidden="true" tabindex="-1"></a><span class="co"># Inner join the tx2gene so we get gene symbols</span></span>
<span id="cb26-234"><a href="#cb26-234" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-235"><a href="#cb26-235" aria-hidden="true" tabindex="-1"></a>t2g <span class="ot">&lt;-</span> grch38 <span class="sc">%&gt;%</span> </span>
<span id="cb26-236"><a href="#cb26-236" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(symbol, ensgene) <span class="sc">%&gt;%</span> </span>
<span id="cb26-237"><a href="#cb26-237" aria-hidden="true" tabindex="-1"></a>  dplyr<span class="sc">::</span><span class="fu">inner_join</span>(grch38_tx2gene) <span class="sc">%&gt;%</span> </span>
<span id="cb26-238"><a href="#cb26-238" aria-hidden="true" tabindex="-1"></a>  <span class="fu">rename</span>(<span class="at">target_id =</span> enstxp, </span>
<span id="cb26-239"><a href="#cb26-239" aria-hidden="true" tabindex="-1"></a>         <span class="at">ens_gene =</span> ensgene, </span>
<span id="cb26-240"><a href="#cb26-240" aria-hidden="true" tabindex="-1"></a>         <span class="at">ext_gene =</span> symbol)</span>
<span id="cb26-241"><a href="#cb26-241" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-242"><a href="#cb26-242" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-243"><a href="#cb26-243" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-244"><a href="#cb26-244" aria-hidden="true" tabindex="-1"></a><span class="fu">### Step 2: Fit the sleuth model</span></span>
<span id="cb26-245"><a href="#cb26-245" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-246"><a href="#cb26-246" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-247"><a href="#cb26-247" aria-hidden="true" tabindex="-1"></a>&lt;img src="../img/sleuth_workflow2_updated.png" width="600"/&gt;</span>
<span id="cb26-248"><a href="#cb26-248" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-249"><a href="#cb26-249" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-250"><a href="#cb26-250" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Fit the transcript abundance data to the Sleuth model</span></span>
<span id="cb26-251"><a href="#cb26-251" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-252"><a href="#cb26-252" aria-hidden="true" tabindex="-1"></a>Using the <span class="in">`sleuth_prep()`</span> function, the counts are normalized and filtered, then merged with the metadata. In addition, the bootstraps for each transcript are summarized. This function can take a bit of time, but there is an option (<span class="in">`ncores`</span>) to split across multiple processors.</span>
<span id="cb26-253"><a href="#cb26-253" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-254"><a href="#cb26-254" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-255"><a href="#cb26-255" aria-hidden="true" tabindex="-1"></a><span class="co"># Create sleuth object for analysis </span></span>
<span id="cb26-256"><a href="#cb26-256" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-257"><a href="#cb26-257" aria-hidden="true" tabindex="-1"></a>so <span class="ot">&lt;-</span> <span class="fu">sleuth_prep</span>(summarydata, </span>
<span id="cb26-258"><a href="#cb26-258" aria-hidden="true" tabindex="-1"></a>                  <span class="at">full_model =</span> design, </span>
<span id="cb26-259"><a href="#cb26-259" aria-hidden="true" tabindex="-1"></a>                  <span class="at">target_mapping =</span> t2g, </span>
<span id="cb26-260"><a href="#cb26-260" aria-hidden="true" tabindex="-1"></a>                  <span class="at">read_bootstrap_tpm =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-261"><a href="#cb26-261" aria-hidden="true" tabindex="-1"></a>                  <span class="at">extra_bootstrap_summary =</span> <span class="cn">TRUE</span>,</span>
<span id="cb26-262"><a href="#cb26-262" aria-hidden="true" tabindex="-1"></a>                  <span class="at">transformation_function =</span> <span class="cf">function</span>(x) <span class="fu">log2</span>(x <span class="sc">+</span> <span class="fl">0.5</span>))  </span>
<span id="cb26-263"><a href="#cb26-263" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-264"><a href="#cb26-264" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-265"><a href="#cb26-265" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;**NOTE:** By default the transformation of counts is natural log, which would make the output fold changes somewhat more difficult to interpret. By specifying the </span><span class="in">`transformation_function`</span><span class="at"> to be </span><span class="in">`log2(x + 0.5)`</span><span class="at"> we are ensuring our output fold changes are log2.</span></span>
<span id="cb26-266"><a href="#cb26-266" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-267"><a href="#cb26-267" aria-hidden="true" tabindex="-1"></a>In fitting the sleuth model, sleuth performs shrinkage of variance, parameter estimation and estimation of variance using the general linear model:</span>
<span id="cb26-268"><a href="#cb26-268" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-269"><a href="#cb26-269" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-270"><a href="#cb26-270" aria-hidden="true" tabindex="-1"></a>so <span class="ot">&lt;-</span> <span class="fu">sleuth_fit</span>(so)</span>
<span id="cb26-271"><a href="#cb26-271" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-272"><a href="#cb26-272" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-273"><a href="#cb26-273" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; **NOTE:** alternatively the two prior steps could have been run as: </span></span>
<span id="cb26-274"><a href="#cb26-274" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```r</span></span>
<span id="cb26-275"><a href="#cb26-275" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; so &lt;- sleuth_prep(summarydata, </span></span>
<span id="cb26-276"><a href="#cb26-276" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;                  full_model = design, </span></span>
<span id="cb26-277"><a href="#cb26-277" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;                  target_mapping = t2g, </span></span>
<span id="cb26-278"><a href="#cb26-278" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;                  read_bootstrap_tpm = TRUE,</span></span>
<span id="cb26-279"><a href="#cb26-279" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;                  extra_bootstrap_summary = TRUE,</span></span>
<span id="cb26-280"><a href="#cb26-280" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;                  transformation_function = function(x) log2(x + 0.5)) %&gt;%</span></span>
<span id="cb26-281"><a href="#cb26-281" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;   sleuth_fit()</span></span>
<span id="cb26-282"><a href="#cb26-282" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; ```</span></span>
<span id="cb26-283"><a href="#cb26-283" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-284"><a href="#cb26-284" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Check which models have been fit and which coefficients can be tested</span></span>
<span id="cb26-285"><a href="#cb26-285" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-286"><a href="#cb26-286" aria-hidden="true" tabindex="-1"></a>Ensure the design model and coefficients are correct for your analysis. The level not shown is the base level.</span>
<span id="cb26-287"><a href="#cb26-287" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-288"><a href="#cb26-288" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-289"><a href="#cb26-289" aria-hidden="true" tabindex="-1"></a><span class="fu">models</span>(so)</span>
<span id="cb26-290"><a href="#cb26-290" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-291"><a href="#cb26-291" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-292"><a href="#cb26-292" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; **NOTE:** Sleuth will automatically use the first level (alphabetically by default) in the factor variable being tested to compare all other conditions against (in our metadata, this is 'control'). If you want to use a different condition to be the base level, then you would need to use the relevel() function to change the base level of the variable in step 1 above. For example, if we wanted the base level of </span><span class="in">`sampletype`</span><span class="at"> to be "MOV10_knockdown", we could use the following code:</span></span>
<span id="cb26-293"><a href="#cb26-293" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;</span></span>
<span id="cb26-294"><a href="#cb26-294" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;```r</span></span>
<span id="cb26-295"><a href="#cb26-295" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; # DO NOT RUN!</span></span>
<span id="cb26-296"><a href="#cb26-296" aria-hidden="true" tabindex="-1"></a><span class="at">&gt; summarydata$sampletype &lt;- relevel(summarydata$sampletype, ref = "MOV10_knockdown")</span></span>
<span id="cb26-297"><a href="#cb26-297" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;```</span></span>
<span id="cb26-298"><a href="#cb26-298" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;***An ordered factor will not give interpretable output, so do NOT order the factor using the factor() function with `ordered = TRUE`, use factor() specifying the levels or relevel() instead.***</span></span>
<span id="cb26-299"><a href="#cb26-299" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-300"><a href="#cb26-300" aria-hidden="true" tabindex="-1"></a><span class="fu">### Step 3: Test significant differences between conditions using the Wald test</span></span>
<span id="cb26-301"><a href="#cb26-301" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-302"><a href="#cb26-302" aria-hidden="true" tabindex="-1"></a>At this step in the workflow, we need to specify which level we want to compare against the base level (use the name given for the coefficients from <span class="in">`models(so)`</span>):</span>
<span id="cb26-303"><a href="#cb26-303" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-304"><a href="#cb26-304" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-305"><a href="#cb26-305" aria-hidden="true" tabindex="-1"></a><span class="co"># Wald test for differential expression of isoforms</span></span>
<span id="cb26-306"><a href="#cb26-306" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-307"><a href="#cb26-307" aria-hidden="true" tabindex="-1"></a>oe <span class="ot">&lt;-</span> <span class="fu">sleuth_wt</span>(so, </span>
<span id="cb26-308"><a href="#cb26-308" aria-hidden="true" tabindex="-1"></a>                <span class="at">which_beta =</span> <span class="st">'sampletypeMOV10_overexpression'</span>)</span>
<span id="cb26-309"><a href="#cb26-309" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-310"><a href="#cb26-310" aria-hidden="true" tabindex="-1"></a><span class="co"># output results</span></span>
<span id="cb26-311"><a href="#cb26-311" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-312"><a href="#cb26-312" aria-hidden="true" tabindex="-1"></a>sleuth_results_oe <span class="ot">&lt;-</span> <span class="fu">sleuth_results</span>(oe, </span>
<span id="cb26-313"><a href="#cb26-313" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">test =</span> <span class="st">'sampletypeMOV10_overexpression'</span>, </span>
<span id="cb26-314"><a href="#cb26-314" aria-hidden="true" tabindex="-1"></a>                                    <span class="at">show_all =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-315"><a href="#cb26-315" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-316"><a href="#cb26-316" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-317"><a href="#cb26-317" aria-hidden="true" tabindex="-1"></a><span class="at">&gt;**NOTE:** </span></span>
<span id="cb26-318"><a href="#cb26-318" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-319"><a href="#cb26-319" aria-hidden="true" tabindex="-1"></a>The output represents the results from the differential expression testing with the following columns:</span>
<span id="cb26-320"><a href="#cb26-320" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-321"><a href="#cb26-321" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`target_id`:** the Ensembl transcript ID</span>
<span id="cb26-322"><a href="#cb26-322" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`pval`:** the Wald test FDR adjusted pvalue using Benjamini-Hochberg</span>
<span id="cb26-323"><a href="#cb26-323" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`qval`:** the p-value adjusted for multiple test correction</span>
<span id="cb26-324"><a href="#cb26-324" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`b`:** beta value, which is the log2 fold changes between conditions (These are log2 b/c we specified log2 transformation in the <span class="in">`sleuth_prep()`</span> step. By default, these would have been natural log fold changes).</span>
<span id="cb26-325"><a href="#cb26-325" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`se_b`:** standard error of the beta value</span>
<span id="cb26-326"><a href="#cb26-326" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`mean_obs`:** the mean expression (log2) of the transcript across all samples</span>
<span id="cb26-327"><a href="#cb26-327" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`var_obs`:** the biological variance of the expression</span>
<span id="cb26-328"><a href="#cb26-328" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`tech_var`:** the technical variance of expression (derived from the bootstraps)</span>
<span id="cb26-329"><a href="#cb26-329" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`sigma_sq`:** raw estimator of the variance once the technical variance has been removed</span>
<span id="cb26-330"><a href="#cb26-330" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`smooth_sigma_sq`:** the smooth regression fit for the shrinkage estimation</span>
<span id="cb26-331"><a href="#cb26-331" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`final_sigma_sq`:** max(sigma_sq, smooth_sigma_sq). this is the one used for covariance estimation of beta (in addition to tech_var)</span>
<span id="cb26-332"><a href="#cb26-332" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`ens_gene`:** associated Ensembl gene ID</span>
<span id="cb26-333"><a href="#cb26-333" aria-hidden="true" tabindex="-1"></a><span class="ss">- </span>**`ext_gene`:** associated gene symbol</span>
<span id="cb26-334"><a href="#cb26-334" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-335"><a href="#cb26-335" aria-hidden="true" tabindex="-1"></a><span class="ss">* </span>Adapted from the results table description in Harold Pimental's <span class="co">[</span><span class="ot">post</span><span class="co">](https://groups.google.com/forum/#!searchin/kallisto-sleuth-users/mean_obs%7Csort:date/kallisto-sleuth-users/kWodd7CQejE/nzr6hEOKAAAJ)</span> to the 'kallisto-sleuth-users' google groups. Harold Pimental is an author of sleuth.*</span>
<span id="cb26-336"><a href="#cb26-336" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-337"><a href="#cb26-337" aria-hidden="true" tabindex="-1"></a><span class="al">![sleuth_results](../img/sleuth_mov10_results.png)</span></span>
<span id="cb26-338"><a href="#cb26-338" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-339"><a href="#cb26-339" aria-hidden="true" tabindex="-1"></a><span class="fu">### Exploring transcript-level expression between samples</span></span>
<span id="cb26-340"><a href="#cb26-340" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-341"><a href="#cb26-341" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Exploratory analyses:</span></span>
<span id="cb26-342"><a href="#cb26-342" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-343"><a href="#cb26-343" aria-hidden="true" tabindex="-1"></a>Now that we have our results, we can perform some exploratory analyses, such as PCA, heatmap, and distributions of counts between conditions. By default, these plots will use the sleuth-normalized <span class="in">`est_counts`</span> (not the log2 transformed values).</span>
<span id="cb26-344"><a href="#cb26-344" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-345"><a href="#cb26-345" aria-hidden="true" tabindex="-1"></a>**PCA:** There are multiple functions to explore the variation in the dataset explained by the different PCs.</span>
<span id="cb26-346"><a href="#cb26-346" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-347"><a href="#cb26-347" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-348"><a href="#cb26-348" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pca</span>(oe, </span>
<span id="cb26-349"><a href="#cb26-349" aria-hidden="true" tabindex="-1"></a>         <span class="at">color_by =</span> <span class="st">'sampletype'</span>,</span>
<span id="cb26-350"><a href="#cb26-350" aria-hidden="true" tabindex="-1"></a>         <span class="at">text_labels =</span> <span class="cn">TRUE</span>)</span>
<span id="cb26-351"><a href="#cb26-351" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-352"><a href="#cb26-352" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-353"><a href="#cb26-353" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-354"><a href="#cb26-354" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/pca_sleuth.png" width="700"/&gt;</span>
<span id="cb26-355"><a href="#cb26-355" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-356"><a href="#cb26-356" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-357"><a href="#cb26-357" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-358"><a href="#cb26-358" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_pc_variance</span>(oe)</span>
<span id="cb26-359"><a href="#cb26-359" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-360"><a href="#cb26-360" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-361"><a href="#cb26-361" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-362"><a href="#cb26-362" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_pca_variance.png" width="500"/&gt;</span>
<span id="cb26-363"><a href="#cb26-363" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-364"><a href="#cb26-364" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-365"><a href="#cb26-365" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-366"><a href="#cb26-366" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_loadings</span>(oe)</span>
<span id="cb26-367"><a href="#cb26-367" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-368"><a href="#cb26-368" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-369"><a href="#cb26-369" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-370"><a href="#cb26-370" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_pca_loadings.png" width="500"/&gt;</span>
<span id="cb26-371"><a href="#cb26-371" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-372"><a href="#cb26-372" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-373"><a href="#cb26-373" aria-hidden="true" tabindex="-1"></a>**Heatmap:** The heatmap plot is shaded with the Jensen-Shannon divergence values. Therefore, **lower divergence values represent samples that are more similar to each other**.</span>
<span id="cb26-374"><a href="#cb26-374" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-375"><a href="#cb26-375" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-376"><a href="#cb26-376" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_sample_heatmap</span>(oe)</span>
<span id="cb26-377"><a href="#cb26-377" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-378"><a href="#cb26-378" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-379"><a href="#cb26-379" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-380"><a href="#cb26-380" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_cor_heatmap.png" width="600"/&gt;</span>
<span id="cb26-381"><a href="#cb26-381" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-382"><a href="#cb26-382" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-383"><a href="#cb26-383" aria-hidden="true" tabindex="-1"></a>**Count distributions:** There is a histogram plot to explore count distributions between sample groups, which should be similar to each other when performing DE testing. The count distributions represent the proportion of genes (on the y-axis) associated with the number of counts (designated on the x-axis):</span>
<span id="cb26-384"><a href="#cb26-384" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-385"><a href="#cb26-385" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-386"><a href="#cb26-386" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_group_density</span>(oe, </span>
<span id="cb26-387"><a href="#cb26-387" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_filtered =</span> <span class="cn">FALSE</span>, </span>
<span id="cb26-388"><a href="#cb26-388" aria-hidden="true" tabindex="-1"></a>                   <span class="at">units =</span> <span class="st">"est_counts"</span>,</span>
<span id="cb26-389"><a href="#cb26-389" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trans =</span> <span class="st">"log"</span>, </span>
<span id="cb26-390"><a href="#cb26-390" aria-hidden="true" tabindex="-1"></a>                   <span class="at">grouping =</span> <span class="st">"sampletype"</span>)</span>
<span id="cb26-391"><a href="#cb26-391" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-392"><a href="#cb26-392" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-393"><a href="#cb26-393" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-394"><a href="#cb26-394" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/plot_density_unfiltered.png" width="600"/&gt;</span>
<span id="cb26-395"><a href="#cb26-395" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-396"><a href="#cb26-396" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-397"><a href="#cb26-397" aria-hidden="true" tabindex="-1"></a>As we know, most genes have few counts, but we filter these genes prior to performing DE analysis. If we want to look at the distributions of the filtered genes used for DE analysis, we could change the <span class="in">`use_filtered`</span> argument to <span class="in">`TRUE`</span>.</span>
<span id="cb26-398"><a href="#cb26-398" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-399"><a href="#cb26-399" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-400"><a href="#cb26-400" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_group_density</span>(oe, </span>
<span id="cb26-401"><a href="#cb26-401" aria-hidden="true" tabindex="-1"></a>                   <span class="at">use_filtered =</span> <span class="cn">TRUE</span>, </span>
<span id="cb26-402"><a href="#cb26-402" aria-hidden="true" tabindex="-1"></a>                   <span class="at">units =</span> <span class="st">"est_counts"</span>,</span>
<span id="cb26-403"><a href="#cb26-403" aria-hidden="true" tabindex="-1"></a>                   <span class="at">trans =</span> <span class="st">"log"</span>, </span>
<span id="cb26-404"><a href="#cb26-404" aria-hidden="true" tabindex="-1"></a>                   <span class="at">grouping =</span> <span class="st">"sampletype"</span>)</span>
<span id="cb26-405"><a href="#cb26-405" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-406"><a href="#cb26-406" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-407"><a href="#cb26-407" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-408"><a href="#cb26-408" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/plot_density_filtered.png" width="600"/&gt;</span>
<span id="cb26-409"><a href="#cb26-409" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-410"><a href="#cb26-410" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-411"><a href="#cb26-411" aria-hidden="true" tabindex="-1"></a><span class="fu">#### Results analyses: </span></span>
<span id="cb26-412"><a href="#cb26-412" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-413"><a href="#cb26-413" aria-hidden="true" tabindex="-1"></a>There are also functions to explore the results. For example, we can perform an expression heatmap for select transcripts:</span>
<span id="cb26-414"><a href="#cb26-414" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-415"><a href="#cb26-415" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-416"><a href="#cb26-416" aria-hidden="true" tabindex="-1"></a>sig_transcripts <span class="ot">&lt;-</span> sleuth_results_oe <span class="sc">%&gt;%</span> </span>
<span id="cb26-417"><a href="#cb26-417" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(qval <span class="sc">&lt;</span> <span class="fl">0.05</span>)</span>
<span id="cb26-418"><a href="#cb26-418" aria-hidden="true" tabindex="-1"></a>  </span>
<span id="cb26-419"><a href="#cb26-419" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_transcript_heatmap</span>(oe, </span>
<span id="cb26-420"><a href="#cb26-420" aria-hidden="true" tabindex="-1"></a>                        <span class="at">transcripts =</span> sig_transcripts<span class="sc">$</span>target_id[<span class="dv">1</span><span class="sc">:</span><span class="dv">20</span>])</span>
<span id="cb26-421"><a href="#cb26-421" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-422"><a href="#cb26-422" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-423"><a href="#cb26-423" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-424"><a href="#cb26-424" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/heatmap_sleuth.png" width="600"/&gt;</span>
<span id="cb26-425"><a href="#cb26-425" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-426"><a href="#cb26-426" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-427"><a href="#cb26-427" aria-hidden="true" tabindex="-1"></a>Sleuth also has some handy functions to plot expression of transcripts with bootstrap variation to **visualize both biological and technical variation** for selected transcripts:</span>
<span id="cb26-428"><a href="#cb26-428" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-429"><a href="#cb26-429" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-430"><a href="#cb26-430" aria-hidden="true" tabindex="-1"></a><span class="co"># Plotting</span></span>
<span id="cb26-431"><a href="#cb26-431" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-432"><a href="#cb26-432" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bootstrap</span>(oe, </span>
<span id="cb26-433"><a href="#cb26-433" aria-hidden="true" tabindex="-1"></a>               <span class="at">target_id =</span> <span class="st">"ENST00000495374"</span>, </span>
<span id="cb26-434"><a href="#cb26-434" aria-hidden="true" tabindex="-1"></a>               <span class="at">units =</span> <span class="st">"est_counts"</span>, </span>
<span id="cb26-435"><a href="#cb26-435" aria-hidden="true" tabindex="-1"></a>               <span class="at">color_by =</span> <span class="st">"sampletype"</span>)</span>
<span id="cb26-436"><a href="#cb26-436" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-437"><a href="#cb26-437" aria-hidden="true" tabindex="-1"></a><span class="fu">plot_bootstrap</span>(oe, </span>
<span id="cb26-438"><a href="#cb26-438" aria-hidden="true" tabindex="-1"></a>               <span class="at">target_id =</span> <span class="st">"ENST00000367412"</span>, </span>
<span id="cb26-439"><a href="#cb26-439" aria-hidden="true" tabindex="-1"></a>               <span class="at">units =</span> <span class="st">"est_counts"</span>, </span>
<span id="cb26-440"><a href="#cb26-440" aria-hidden="true" tabindex="-1"></a>               <span class="at">color_by =</span> <span class="st">"sampletype"</span>)</span>
<span id="cb26-441"><a href="#cb26-441" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-442"><a href="#cb26-442" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-443"><a href="#cb26-443" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-444"><a href="#cb26-444" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_bootstraps1.png" width="600"/&gt;</span>
<span id="cb26-445"><a href="#cb26-445" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-446"><a href="#cb26-446" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-447"><a href="#cb26-447" aria-hidden="true" tabindex="-1"></a>&lt;p align="center"&gt;</span>
<span id="cb26-448"><a href="#cb26-448" aria-hidden="true" tabindex="-1"></a>  &lt;img src="../img/sleuth_bootstraps2.png" width="600"/&gt;</span>
<span id="cb26-449"><a href="#cb26-449" aria-hidden="true" tabindex="-1"></a>&lt;/p&gt;</span>
<span id="cb26-450"><a href="#cb26-450" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-451"><a href="#cb26-451" aria-hidden="true" tabindex="-1"></a>Sleuth also offers us the option to explore the data and results interactively using a web interface. </span>
<span id="cb26-452"><a href="#cb26-452" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-453"><a href="#cb26-453" aria-hidden="true" tabindex="-1"></a><span class="in">```r</span></span>
<span id="cb26-454"><a href="#cb26-454" aria-hidden="true" tabindex="-1"></a><span class="fu">sleuth_live</span>(oe)</span>
<span id="cb26-455"><a href="#cb26-455" aria-hidden="true" tabindex="-1"></a><span class="in">```</span></span>
<span id="cb26-456"><a href="#cb26-456" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-457"><a href="#cb26-457" aria-hidden="true" tabindex="-1"></a>Using this web interface, we can explore diagnostic plots and summaries of experimental factors and expression levels. We also have the ability to perform clustering analyses such as PCA and heatmaps. Finally, we can analyze the differential expression results by plotting MA and volcano plots and by exploring expression levels at the transcript and gene levels. A brief tutorial explaining the options available interactively can be found <span class="co">[</span><span class="ot">here</span><span class="co">](http://www.rna-seqblog.com/iihg-intro-to-sleuth-for-rna-seq/)</span>.</span>
<span id="cb26-458"><a href="#cb26-458" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-459"><a href="#cb26-459" aria-hidden="true" tabindex="-1"></a>Explore the results table under the <span class="in">`analyses`</span> tab, <span class="in">`test table`</span>.</span>
<span id="cb26-460"><a href="#cb26-460" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-461"><a href="#cb26-461" aria-hidden="true" tabindex="-1"></a><span class="al">![results_table](../img/mov10_test_table.png)</span></span>
<span id="cb26-462"><a href="#cb26-462" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-463"><a href="#cb26-463" aria-hidden="true" tabindex="-1"></a>Look at the expression levels of Mov10 for three different isoforms using the <span class="in">`analyses`</span> tab, <span class="in">`gene view`</span>. Look for the Mov10 gene, ENSG00000155363. Compare expression of the different isoforms of Mov10.</span>
<span id="cb26-464"><a href="#cb26-464" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-465"><a href="#cb26-465" aria-hidden="true" tabindex="-1"></a><span class="al">![mov_isoforms](../img/mov10_isoform_expression.png)</span></span>
<span id="cb26-466"><a href="#cb26-466" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb26-467"><a href="#cb26-467" aria-hidden="true" tabindex="-1"></a>***</span>
<span id="cb26-468"><a href="#cb26-468" aria-hidden="true" tabindex="-1"></a>*This lesson has been developed by members of the teaching team at the [Harvard Chan Bioinformatics Core (HBC)](http://bioinformatics.sph.harvard.edu/). These are open access materials distributed under the terms of the [Creative Commons Attribution license](https://creativecommons.org/licenses/by/4.0/) (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.*</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div></div></div></div></div>
</div> <!-- /content -->
<footer class="footer">
  <div class="nav-footer">
    <div class="nav-footer-left">
      &nbsp;
    </div>   
    <div class="nav-footer-center">
<p>This lesson has been developed by members of the teaching team at the <a href="http://bioinformatics.sph.harvard.edu/">Harvard Chan Bioinformatics Core (HBC)</a> RRID:SCR_025373. <br> These are open access materials distributed under the terms of the <a href="https://creativecommons.org/licenses/by/4.0/">Creative Commons Attribution license</a> (CC BY 4.0), which permits unrestricted use, distribution, and reproduction in any medium, provided the original author and source are credited.</p>
</div>
    <div class="nav-footer-right">
      &nbsp;
    </div>
  </div>
</footer>




</body></html>