<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.555">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Meeta Mistry, Radhika Khetani, Mary Piper">
<meta name="dcterms.date" content="2017-10-24">

<title>DGE Analysis Homework</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="DGE_analysis_exercises_files/libs/clipboard/clipboard.min.js"></script>
<script src="DGE_analysis_exercises_files/libs/quarto-html/quarto.js"></script>
<script src="DGE_analysis_exercises_files/libs/quarto-html/popper.min.js"></script>
<script src="DGE_analysis_exercises_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="DGE_analysis_exercises_files/libs/quarto-html/anchor.min.js"></script>
<link href="DGE_analysis_exercises_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="DGE_analysis_exercises_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="DGE_analysis_exercises_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="DGE_analysis_exercises_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="DGE_analysis_exercises_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">


<link rel="stylesheet" href="css/styles.css">
</head>

<body>

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">
<div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
  <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#using-deseq2-for-gene-level-differential-expression-analysis" id="toc-using-deseq2-for-gene-level-differential-expression-analysis" class="nav-link active" data-scroll-target="#using-deseq2-for-gene-level-differential-expression-analysis">Using DESeq2 for gene-level differential expression analysis</a></li>
  <li><a href="#working-with-the-deseq2-results-table" id="toc-working-with-the-deseq2-results-table" class="nav-link" data-scroll-target="#working-with-the-deseq2-results-table">Working with the DESeq2 results table</a></li>
  <li><a href="#visualizing-results" id="toc-visualizing-results" class="nav-link" data-scroll-target="#visualizing-results">Visualizing results</a></li>
  <li><a href="#use-significant-gene-lists-to-find-overlaps-between-the-two-comparisons" id="toc-use-significant-gene-lists-to-find-overlaps-between-the-two-comparisons" class="nav-link" data-scroll-target="#use-significant-gene-lists-to-find-overlaps-between-the-two-comparisons">Use significant gene lists to find overlaps between the two comparisons</a></li>
  <li><a href="#using-salmon-abundance-estimates-with-deseq2" id="toc-using-salmon-abundance-estimates-with-deseq2" class="nav-link" data-scroll-target="#using-salmon-abundance-estimates-with-deseq2">Using Salmon abundance estimates with DESeq2</a></li>
  </ul>
</nav>
</div>
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">DGE Analysis Homework</h1>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Meeta Mistry, Radhika Khetani, Mary Piper </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">October 24, 2017</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="using-deseq2-for-gene-level-differential-expression-analysis" class="level1">
<h1>Using DESeq2 for gene-level differential expression analysis</h1>
<p>The metadata below describes an RNA-seq analysis experiment, in which the metadata table below and associated count matrix have been loaded into R as <code>meta</code> and <code>counts</code>, respectively. Additionally, all of the appropriate libraries have been loaded for you. Use the information in the table to answer the following questions.</p>
<p><strong>meta</strong></p>
<table class="caption-top table">
<thead>
<tr class="header">
<th></th>
<th>Genotype</th>
<th>Celltype</th>
<th>Batch</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td>sample1</td>
<td>Wt</td>
<td>typeA</td>
<td>second</td>
</tr>
<tr class="even">
<td>sample2</td>
<td>Wt</td>
<td>typeA</td>
<td>second</td>
</tr>
<tr class="odd">
<td>sample3</td>
<td>Wt</td>
<td>typeA</td>
<td>first</td>
</tr>
<tr class="even">
<td>sample4</td>
<td>KO</td>
<td>typeA</td>
<td>first</td>
</tr>
<tr class="odd">
<td>sample5</td>
<td>KO</td>
<td>typeA</td>
<td>first</td>
</tr>
<tr class="even">
<td>sample6</td>
<td>KO</td>
<td>typeA</td>
<td>second</td>
</tr>
<tr class="odd">
<td>sample7</td>
<td>Wt</td>
<td>typeB</td>
<td>second</td>
</tr>
<tr class="even">
<td>sample8</td>
<td>Wt</td>
<td>typeB</td>
<td>first</td>
</tr>
<tr class="odd">
<td>sample9</td>
<td>Wt</td>
<td>typeB</td>
<td>second</td>
</tr>
<tr class="even">
<td>sample10</td>
<td>KO</td>
<td>typeB</td>
<td>first</td>
</tr>
<tr class="odd">
<td>sample11</td>
<td>KO</td>
<td>typeB</td>
<td>first</td>
</tr>
<tr class="even">
<td>sample12</td>
<td>KO</td>
<td>typeB</td>
<td>second</td>
</tr>
</tbody>
</table>
<div class="callout callout-style-default callout-note callout-titled">
<div class="callout-header d-flex align-content-center">
<div class="callout-icon-container">
<i class="callout-icon"></i>
</div>
<div class="callout-title-container flex-fill">
Note
</div>
</div>
<div class="callout-body-container callout-body">
<p>This is an exercise in thinking about running DESeq2. <strong>You do not need to run any code in R/RStudio.</strong> Refer to the materials/lessons from class to answer the following questions.</p>
</div>
</div>
<p><strong>a.</strong> Reorder the columns of the <code>counts</code> dataset such that <code>rownames(meta) == colnames(counts)</code>.</p>
<p><strong>b.</strong> Provide the line of code used to create a DESeqDataSet object called <code>dds</code> in which <code>Genotype</code> is the factor of interest and <code>Celltype</code> and <code>Batch</code> are other contributing sources of variation in your data.</p>
<p><strong>c.</strong> Provide the line of code required to run DESeq2 on <code>dds</code>.</p>
<p><strong>d.</strong> Provide the line of code to create a dispersion plot.</p>
<p><strong>e.</strong> Provide the line of code to return the results of a Wald test comparison for <code>Celltype</code> categories <code>typeA</code> versus <code>typeB</code> (i.e., the fold changes reported should reflect gene expression changes relative to <code>typeB</code>).</p>
<p><strong>f.</strong> Provide the line of code to return the results with log2 fold change shrinkage performed.</p>
<p><strong>g.</strong> Provide the line of code to write the results of the Wald test with shrunken log2 fold changes to file.</p>
<p><strong>h.</strong> Provide the line of code to subset the results to return those genes with adjusted p-value &lt; 0.05 and logFold2Change &gt; 1.</p>
</section>
<section id="working-with-the-deseq2-results-table" class="level1">
<h1>Working with the DESeq2 results table</h1>
<p>Using the <code>de_script.R</code> that we created in class for the differential expression analysis, change the thresholds for adjusted p-value and log2 fold change to the following values:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.01</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fl">1.5</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>Using these new cutoffs, perform the following steps:</p>
<p><strong>a.</strong> Subset <code>res_tableOE</code> to only return those rows that meet the criteria we specified above (adjusted p-values &lt; 0.01 and log fold changes &gt;1.5). Save the subsetted table to a data frame called <code>sig_table_hw_oe</code>.</p>
<p><strong>b.</strong> There is a a DESeq2 function that summarizes how many genes are up- and down-regulated using our criteria for <code>alpha=0.01</code>. Use this on the <code>sig_table_hw_oe</code>. Write the code you would use and also list how many genes are up- and down-regulated.</p>
<p><strong>c.</strong> Get the gene names from <code>sig_table_hw_oe</code> and save them to a vector called <code>sigOE_hw</code>.</p>
<p><strong>d.</strong> Write the <code>sigOE_hw</code> vector of gene names to a file called <code>sigOE_hw.txt</code> using the <code>write()</code> function. Ensure the genes are listed in a single column.</p>
</section>
<section id="visualizing-results" class="level1">
<h1>Visualizing results</h1>
<p>For the genes that are differentially expressed in the knockdown versus control comparison (<code>res_tableKD</code>), plot an expression heatmap using normalized counts and <code>pheatmap()</code> following the instructions below. Write the code you would use to create the heatmap.</p>
<p><strong>a.</strong> The heatmap should only include control and knockdown samples.</p>
<p><strong>b.</strong> Set up a heat.colors vector using a palette of your choice from brewer.pal (make sure it is different from the one used in class).</p>
<p><strong>c.</strong> Plot the heatmap without clustering the columns.</p>
<p><strong>d.</strong> Scale expression values by row.</p>
</section>
<section id="use-significant-gene-lists-to-find-overlaps-between-the-two-comparisons" class="level1">
<h1>Use significant gene lists to find overlaps between the two comparisons</h1>
<p>Using the original cutoff values, perform the following steps:</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a>padj.cutoff <span class="ot">&lt;-</span> <span class="fl">0.05</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a>lfc.cutoff <span class="ot">&lt;-</span> <span class="fu">log2</span>(<span class="fl">1.5</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p><strong>a.</strong> Create separate vectors with gene names for up-regulated genes and down-regulated genes from <code>res_tableOE</code> and save as <code>up_OE</code> and <code>down_OE</code>, respectively. Write the code below:</p>
<p><strong>b.</strong> Create separate vectors with gene names for up-regulated genes and down-regulated genes from <code>res_tableKD</code> and save as <code>up_KD</code> and <code>down_KD</code>, respectively. Write the code below:</p>
<p><strong>c.</strong> Test for overlaps between the lists:</p>
<ul>
<li>How many and which genes in <code>up_OE</code> are also in <code>down_KD</code>?</li>
<li>How many and which genes in <code>up_KD</code> are also in <code>down_OE</code>?</li>
</ul>
</section>
<section id="using-salmon-abundance-estimates-with-deseq2" class="level1">
<h1>Using Salmon abundance estimates with DESeq2</h1>
<p>We have materials for using counts from quasialignment tools, such as Salmon, to perform the DE analysis. The <code>tximport</code> package in R is needed to summarize the transcript-level estimates generated from Salmon into pseudocounts.</p>
<p>Open up RStudio and create a project for using Salmon estimates with DESeq2 (i.e., ~/Desktop/salmon should be your working directory).</p>
<p><strong>a.</strong> Follow the <a href="https://github.com/hbctraining/Intro-to-rnaseq-hpc-orchestra/blob/master/lessons/DE_analysis.md#differential-expression-analysis-using-pseudocounts">markdown</a> for the Salmon section to create the DESeqDataSet object.</p>
<p><strong>b.</strong> Run DESeq2 on the DESeqDataSet object (<code>dds</code>) you created.</p>
<p><strong>c.</strong> Use the <code>results()</code> function to extract a results table for the OE vs Ctl comparison. Be sure to provide the appropriate contrasts and to perform shrinkage.</p>
<p><strong>d.</strong> Use the <code>results()</code> function to extract a results table for the KD vs Ctl comparison. Be sure to provide the appropriate contrasts and to perform shrinkage.</p>
<p><strong>e.</strong> Use the <code>summary()</code> function using <code>alpha=0.05</code> on the OE results table.</p>
<ul>
<li><p>Report the number of genes that are up- and down-regulated.</p></li>
<li><p>Report how many genes are removed due to independent filtering. (HINT: these are genes that are filtered due to low mean count.)</p></li>
<li><p>Report how many genes were removed due to an extreme outlier count. (HINT: these ‘outliers’ are identified based on Cook’s distance.)</p></li>
</ul>
<p><strong>f.</strong> Use the <code>summary()</code> function using <code>alpha=0.05</code> on the KD results table.</p>
<ul>
<li><p>Report the number of genes that are up- and down-regulated.</p></li>
<li><p>Report how many genes are removed due to independent filtering. (HINT: these are genes that are filtered due to low mean count.)</p></li>
<li><p>Report how many genes were removed due to an extreme outlier count. (HINT: these ‘outliers’ are identified based on Cook’s distance.)</p></li>
</ul>
<p><strong>g.</strong> Create a vector called <code>criteria_oe</code>, which contains the indexes for which rows in the OE results table have <code>padj &lt; 0.05</code> AND <code>abs(logFC) &gt; log2(1.5)</code>.</p>
<p><strong>h.</strong> Subset the OE results into a new table using the <code>criteria_oe</code> vector.</p>
<p><strong>i.</strong> Create a vector called <code>criteria_kd</code>, which contains the indexes for which rows in the KD results table have <code>padj &lt; 0.05</code> AND <code>abs(logFC) &gt; log2(1.5)</code>.</p>
<p><strong>j.</strong> Subset the KD results into a new table using the <code>criteria_kd</code> vector.</p>
<p><strong>k.</strong> Use the following linked datasets as the significant genes using a different workflow (STAR alignment + featureCounts + DESeq2), but same samples. Download them via these links:</p>
<p><a href="https://wiki.harvard.edu/confluence/download/attachments/216318985/Mov10_oe_2017.txt?version=1&amp;modificationDate=1498507515000&amp;api=v2">Download OE genes (870 genes)</a></p>
<p><a href="https://wiki.harvard.edu/confluence/download/attachments/216318985/Mov10_kd_2017.txt?version=1&amp;modificationDate=1498507515000&amp;api=v2">Download KD genes (689 genes)</a></p>
<p>Load these gene lists in as <code>star_KD</code> and <code>star_OE</code>, using the <code>scan()</code> function in R.</p>
<p><strong>l.</strong> Find overlapping OE genes between those that you have identified in your subsetted results table (using the Salmon abundance estimates) to the <code>star_OE</code> gene list.</p>
<ul>
<li>How many genes overlap?</li>
</ul>
<p><strong>m.</strong> Find overlapping KD genes between those that you have identified in your subsetted results table (using the Salmon abundance estimates) to the <code>star_KD</code> gene list.</p>
<ul>
<li>How many genes overlap?</li>
</ul>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>